\newpage
\section{Multi-trait GWAS with LiMMBo}
In order to show the utility of \gls{limmbo} for joint high-dimensional phenotype analyses and to demonstrate the advantages over single-trait approaches, I analysed the imputed dataset both with \gls{stgwas} and \gls{mtgwas}. 

\subsection{LiMMBo increases power in detecting genetic associations}
\label{subsection:power-yeast}
For both analyses, I used a \gls{lmm} where the sample-by-sample component of the random genetic effect is based on the \gls{rrm}. To obtain an estimate of the \gls{rrm} I first pruned the genome-wide \glspl{snp} (\num{11623}) for \glspl{snp} that are in \gls{ld} within a window of 3kb and show a correlation \(r^2 > 0.2\) . As the dataset is based on an F2 cross, \gls{ld} structure estimation is not straight-forward and this window size is a simple estimate derived from a study on the population genomics of domestic and wild yeasts \citep{Liti2009}. The \gls{ld} pruning reduced the \gls{snp} set for \gls{rrm} estimation to \num{4105} \glspl{snp}. The \gls{rrm} was estimated using the method introduced by \citet{Yang2011} (\cref{subsubsection:grm}). Both \gls{rrm} estimation and \gls{ld} pruning were done via \textit{plink} \citep{Chang2015}.

The first step in the \gls{mtgwas} is the trait-by-trait covariance estimation via \gls{limmbo}. \num{1000} bootstraps of \num{10} traits each were run and their trait-by-trait covariance estimated. The combined trait-by-trait covariance estimates \tmatsub{C}{g} and \tmatsub{C}{n} were used as input estimates for the second step in the \gls{mtgwas}, the \gls{mvlmm} (\cref{eq:lmm-mv}) across all genome-wide \glspl{snp}. I used a \gls{mvlmm} with a trait-design matrix corresponding to the any effect test, i.e. testing for an effect of each \gls{snp} on any of the traits compared versus the null hypothesis of no association (\cref{subsubsection:model-design}).

For the \gls{stgwas}, the trait-by-trait components of the random effects are point estimates (\(\sigma_g\) and \(\sigma_n\)) derived within the \gls{lmm} framework and do not require \textit{a priori} estimation. The \gls{stgwas} was performed for each trait separately, applying univariate\glspl{lmm} (\cref{eq:lmm-uv}) to test the effect of a \gls{snp} on each individual trait. To account for the number of univariate tests, the p-values obtained from the \gls{stgwas} were adjusted for multiple testing by the effective number of conducted tests \(M_\text{eff}\). \(M_\text{eff}\) was introduced by \citet{Galwey2009} and adjusts for multiple testing in a manner similar to the Bonferroni method (\cref{subsection:multiple-testing}, \citep{Dunn1961}). However, it is less conservative, as it does not adjust for total number of tests, but the estimated, effective number of tests, taking correlation between the 
variables and tests into account:
\begin{equation}
 M_\text{eff} = \frac{(\sum^M_{i=1} \sqrt{\lambda_i})^2}{\sum^M_{i=1}\lambda_i},
 \label{eq:meff}
\end{equation}
 where \(\lambda\) are the eigenvalues of the phenotypes' correlation matrix. To adjust for multiple testing in the \gls{stgwas}, the p-values are multiplied with \(M_\text{eff}\) and set to one if the multiplication leads to values greater than one. \(M_\text{eff}\) for the \num{41} growth traits was estimated to be \num{33}. 
 
In order to assess the significance of the single-trait and multi-trait analyses, I followed approaches of previous association studies in yeast crosses \citep{Brem2002,Brem2005,Ehrenreich2010}, where permutations were used to estimate empirical significance levels. With a conservative, theoretical significance threshold of \(p_\text{t}=10^{-5}\), at most one \gls{snp} is expected to be false positive in a total of \(s = 11,623\) \glspl{snp}. To find the empirical \gls{fdr} corresponding to this threshold, I generated \(k = 50\) permutations of the genotypes and fitted the\glspl{lmm} against these permutations. These p-values were used as the empirical p-value distribution and for \(p_\text{t}=10^{-5}\), empirical \glspl{fdr} estimated as 
\(\text{FDR}_{\text{mtGWAS}} =1.2 \times 10^{-5}\) and \(\text{FDR}_{\text{stGWAS}} =8.6 \times 10^{-6}\).
%The p-values obtained from these analyses were then combined and sorted in increasing order. The p-value observed at position \(p_\text{t} \times s \times k\) is used as the empirical FDR. 

\Cref{fig:GWAS-yeast} shows the manhattan plot of the multi-trait and single-trait \gls{gwas}. On several chromosomes (e.g. chr1, chr6 and chr15), \gls{mtgwas} peaks (blue) are observed whereas no \gls{stgwas} peaks (orange; minimum p-value per \gls{snp} across all \num{41} \gls{stgwas}, adjusted for multiple testing) can be detected. On the other hand, there are a few significant loci for the \gls{stgwas} where the multi-trait analyses either does not reach significance (e.g. on chromosome \num{7}) or does not detect any association (e.g. on chromosome \num{4}). For these loci, the underlying genetics seem to be trait specific to magnesium sulfate and hydroquinone, respectively (\cref{fig:stGWAS-yeast} in the appendix). Testing with a \num{41} degrees of freedom test as done in the \gls{mtgwas} hinders the detection of these strong mono-trait associations (compare distributions in \cref{fig:GWAS-stats}) and confirms previous studies showing that the single-trait model for uncorrelated traits is more powerful \citep{Korte2010}. Both, the gain in power for the multi-trait associations and the burden of a multivariate test when the underlying effect is univariate confirm the results obtained from the theoretical power analysis (\cref{section:power-limmbo}). 

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter3/Figures/manhattanplot.png}
	\caption[\textbf{Manhattan plot of p-values from single-trait and multi-trait GWAS.}]{\textbf{Manhattan plot of p-values from single-trait and multi-trait GWAS.} The \gls{stgwas} p-values were adjusted for multiple testing by the effective number of tests (\(M_\text{eff} = 33\)) and only the minimum adjusted p-values across all \num{41} traits per \gls{snp} are shown. The significance line is drawn at the empirical \(\text{FDR}_{\text{stGWAS}} =8.6 \times 10^{-6}\).}
 	\label{fig:GWAS-yeast}
\end{figure}

To quantify the increase in power, I counted the number of \glspl{snp} detected above the permutation-based significant thresholds for both the \gls{stgwas} and the \gls{mtgwas}. Since the number of \glspl{snp} per locus is not constant (based on \gls{ld} structure in the F2 cross and genotyping parameters), I needed a locus-based rather than a \gls{snp}-based count for a fair comparison of the two methods. In order to filter \glspl{snp} based on locus, I used \textit{plink} for \gls{ld} pruning of the \glspl{snp}, choosing a strict threshold of \(r^2 > 0.2\) and increasing \gls{ld} window sizes ranging from \numrange{3}{100}kb.  The maximal \gls{ld} window of \num{100}kb covers between \num{6}\% (chromosome \num{4}) and \num{43}\% (chromosome 1) of total chromosome length (ScerevisaeR64-1-1, ensembl release 90, \citep{Aken2016}). \Cref{tab:sigsnps} shows that the increase in power is present from narrow to broad \gls{ld} pruning, with on average \num{29}\% more significant loci in \gls{mtgwas}.

% Table generated by Excel2LaTeX from sheet 'LDGwasYeast'
\begin{table}[htbp]
  \centering
  \caption[\textbf{Comparison of significant loci in single-trait and multi-trait GWAS.}]{\textbf{Comparison of significant loci in single-trait and multi-trait GWAS.} In the column `All \glspl{snp}', the absolute number of \glspl{snp} beyond the significance threshold for multi-trait and single-trait \gls{gwas} as well as their ratio (multi-trait:single-trait) are depicted. In order to limit the potential bias in the counting of the loci, introduced by different degrees of \gls{ld} for different loci, the genome-wide \glspl{snp} were \gls{ld} pruned and the ratio of significant \glspl{snp} determined for five different \gls{ld} window sizes. }
\begin{tabular}{lllllll}
    \toprule
          & \multicolumn{1}{c}{\multirow{2}[4]{*}{All \glspl{snp}}} & \multicolumn{5}{c}{\gls{ld} pruned with $r^2 \ge 0.2$} \\
\cmidrule{3-7}          &       & \multicolumn{1}{r}{\num{3}kb} & \multicolumn{1}{r}{\num{10}kb} & \multicolumn{1}{r}{\num{30}kb} & \multicolumn{1}{r}{\num{50}kb} & \multicolumn{1}{r}{\num{100}kb} \\
    \midrule
    NrSNPs & \num{11623} & \num{4105} & \num{1028} & \num{264} & \num{161} & \num{107} \\
    multitrait & \num{1132} & \num{384} & \num{101} & \num{24} & \num{15} & \num{9} \\
    singletrait & \num{695} & \num{275} & \num{72} & \num{20} & \num{13} & \num{7} \\
    multitrait:singletrait & \num{1.63} & \num{1.4} & \num{1.4} & \num{1.2} & \num{1.15} & \num{1.29} \\
    \bottomrule
    \end{tabular}%
  \label{tab:sigsnps}%
\end{table}%

\subsection{Multi-trait effect size estimates as indicators for common biology}
As well as providing an increase in power, the \gls{mtgwas} inherently provides effect size estimates across all phenotypes for a particular locus, allowing for a richer exploration of pleiotropic effects of each of locus.
To analyse the relationship between traits and \glspl{snp} based on their effect size estimates, I filtered the genome-wide \glspl{snp} for \glspl{snp} that fell within a gene body and pruned these \num{8135} \glspl{snp} for \glspl{snp} in \gls{ld}  with \(r^2 > 0.2\) and within a \num{3}kb window (\num{1412} \glspl{snp}). Lastly, I filtered for \glspl{snp} passing the \(\text{FDR}=10^{-5}\) yielding \num{210} \glspl{snp} across \num{15} out of the \num{16} yeast chromosomes. Chromosome \num{5} is the only chromosome without significantly associated \glspl{snp} in the single-trait and multi-trait \gls{gwas} (\cref{fig:GWAS-yeast}). 

To find groups of \glspl{snp} and traits with similar effect size estimates, I clustered the effect size estimates of these \glspl{snp} both across traits and \glspl{snp}. (\cref{fig:effectsizes}). I used the hierarchical clustering algorithm \textit{pvclust} that provides bootstrap-based p-values as a measure for the stability of a given cluster \citep{Suzuki2006}. The clustering was based on their Pearson correlation coefficients, with \num{50000} bootstraps for traits and \num{10000} for \glspl{snp}. Clusters with \(p < 0.05\) were considered significant. A heatmap of effect size estimates and the clustering results is depicted in \cref{fig:effectsizes}. Ignoring the clustering for a first impression of the results, one can clearly see that most \glspl{snp} have significant, non-zero effects in more than one trait (\cref{fig:effectsizes}, strong signals across columns). Furthermore some traits have contributions from across the genome, many of which are xenobiotic growth conditions e.g. zeocin \citep{Krol2015} and neomycin \citep{Foiani2010}. Turning to the clustering, \cref{fig:effectsizes} (dendrograms) shows that the clusters are driven by specific combinations of loci and traits, and would be hard to achieve from a single-trait analysis. 

There are a number of significant clusters of traits (\cref{fig:effectsizes}, blue branches in the row dendrogram), including classically linked carbon metabolism sources (lactose, lactate and ethanol), and other clusters for which there is literature support. For example, expression of genes involved in DNA replication has been shown to change upon treatment with hydroxyurea and \num{4}-nitroquinoline-l-oxide (x4NQO) \citep{Elledge1990}, two substances that are linked in this analysis by forming a stable cluster. A study demonstrating trehalose and sorbitol to have synergistic effects on viability in yeast \citep{Hua2015} demonstrating a biological link of these sugars forming a cluster. For other clusters, such as SDS and Hydroxybenzaldehyde or magnesium sulfate and berbamine I was unable to find literature support. However, these could serve as candidate clusters for further investigation of growth phenotypes.

I discovered \num{31} stable \gls{snp} clusters (\cref{fig:effectsizes}, blue branches in the column dendrogram), many of which represent linked loci. However, there are nine clusters (\cref{fig:effectsizes}, grey boxes) spanning multiple chromosomes, and many clusters linking disjoint regions across a chromosome. Some \gls{snp} clusters have suggestive common annotation, such as \textit{cluster a} which has two members of the nuclear pore complex (NUP1, NUP188), and \textit{cluster b} which has a common set of vesicle associated genes (ATG5, PXA1,VPS41; \cref{fig:effectsizes}, labelled boxes). The small size of the clusters prevented any systematic gene ontology based enrichment. Nevertheless, the ability to explore clusters of both traits and genetic loci demonstrate the utility of \gls{mtgwas} for hypothesis generation.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 150mm 0mm 0mm, clip, width=\textwidth]{Chapter3/Figures/effectsizes.png}
	\caption[\textbf{Hierarchical clustering of mtGWAS effects size estimates.}]{\textbf{Hierarchical clustering of mtGWAS effects size estimates.} Effect size estimates of \gls{ld} pruned (\num{3}kb window, \(r^2 > 0.2\)), significant \glspl{snp} located within a gene body were clustered by loci and traits (both hierarchical, average-linkage clustering of Pearson correlation coefficients ). Stable clusters (pvclust \( p < 0.05\)) are marked in blue. Grey boxes indicate stable \gls{snp} clusters spread across at least two chromosomes. a and b label two clusters for which suggestive common annotation was found, for details see text.}
 	\label{fig:effectsizes}
\end{figure}

\section{Summary}
A particular benefit of \glspl{lmm} is that complex genetic relationships can be modelled, which is useful in structured populations such as this \(F_2\) cross in yeast. In univariate \glspl{lmm}, the kinship information is used to account for background genetic effects in associations with a single trait. When used in \glspl{mvlmm}, the kinship structure allows for the estimation of complex trait covariance structure. However, it is only possible through a combination of appropriate phenotype imputation and a method like \gls{limmbo} to efficiently map all \num{41} growth traits together in order to investigate pleiotropic effects on a genome-wide level. I demonstrated that such a multivariate analysis through \gls{limmbo} is more powerful in detecting genetic associations in a real dataset, than univariate tests. While the focus of this chapter was to demonstrate the applicability and power of \gls{limmbo}, it also highlighted the potential of multivariate analysis for gaining insights into the underlying biology of pleiotropic loci. The effect sizes estimated by the multivariate \gls{lmm} provide the relevant data to study shared pathways and regulation and can help to generate hypotheses for future research.
