\section{Left ventricular trabeculation is associated with two genomic loci}
The extraction of FD measurements from the 2D CMR images yields quantitative phenotypes capturing the complexity of trabeculation in the left ventricle. I used the two summary measures \(\text{FD}_\text{max}^\text{basal}\) and \(\text{FD}_\text{max}^\text{apical}\) described above as the response variables in a multi-trait GWAS with the genetic marker and sex, age, height and weight as covariates. Since the dataset contained related individuals, I extended to model used in \cref{section:GWAS-3Dheart} to a LMM by including an additional random genetic effect based on the RRM of the samples. The RRM was estimated from the samples' genotypes as described in \cref{subsubsection:grm}. The manhattan and qq-plots for the joint analysis of \(\text{FD}_\text{max}^\text{basal}\) and \(\text{FD}_\text{max}^\text{apical}\) are depicted in \cref{fig:manhattan-FD} and \cref{fig:qq-FD}, showing two loci that reach genome-wide significance. As a comparison, single-trait GWAS of  \(\text{FD}_\text{max}^\text{basal}\) and \(\text{FD}_\text{max}^\text{apical}\) only discovered the significant association on chromosome 2 (with response variable \(\text{FD}_\text{max}^\text{apical}\); \cref{fig:manhattan-FD-single}), demonstrating the power of the multi-trait approach.
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter6/Figures/lm_mt_pcs_manhattanplot.png}
	\caption[\textbf{Manhattan plot of multi-trait GWAS on left ventricular trabeculation. }]{\textbf{Manhattan plot of multi-trait GWAS on left ventricular trabeculation. } The maximal apical and basal FD  were modelled jointly in an any effect multi-trait GWAS. The p-values of all genome-wide SNPs are depicted. The horizontal grey line is drawn at the level of genome-wide significance: \(p = 5 \times 10^{-8}\).} 
	 	\label{fig:manhattan-FD}
\end{figure}
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.6\textwidth]{Chapter6/Figures/lm_mt_pcs_qqplot.png}
	\caption[\textbf{Quantile-quantile plot of multi-trait GWAS on left ventricular trabeculation .}]{\textbf{Quantile-quantile plot of multi-trait GWAS on left ventricular trabeculation.} The observed genome-wide p-values of the multi-trait FD GWAS are plotted against p-values drawn from a uniform distribution in \([0,1]\) of the same sample size (expected p-values). The diagonal line represents The diagonal line starts at the origin and has slope one.} 
	 	\label{fig:qq-FD}
\end{figure}
%
\noindent A summary of the two loci that reach genome-wide significance is shown in \cref{tab:sig-FD} and \cref{fig:locuszoom-fd}. The locus on chromosome~2 lies within an intron of a long intergenic noncoding RNAs (lincRNA) of unknown function (\cref{fig:locuszoom-fd}, upper panel). The second associated locus is positioned in intron \num{24} of the ADAMTSL1 gene (\cref{fig:locuszoom-fd}, lower panel). ADAMTSL1 is also known as Punctin and two of its intronic and intergenic variants  (rs7869627: intron \num{17}; rs1411242: intergenic between SH3GL2 and ADAMTSL1) have been found associated with blood pressure phenotypes \citep{Sabatti2009}. rs7855681 is in weak LD with rs7869627 ( \(r^2=0.119\)). 

% Table generated by Excel2LaTeX from sheet 'AssociationFD'
\begin{table}[htbp]
  \centering
  \caption[\textbf{SNPs with strongest association in left ventricular trabeculation GWAS. }]{\textbf{SNPs with strongest association in left ventricular trabeculation GWAS. } For each significant locus, the p-values for SNPs in LD with an \(r^2 > 0.8\) in a \num{50}kb window were compared and only the SNP with smallest p-value per locus listed below. M allele: major allele, m allele:  minor allele, MAF: minor allele frequency. }
    \begin{tabular}{lrllll}
    \toprule
    SNP   & \multicolumn{1}{l}{Chr} & Position & P-value & M/m allele & MAF \\
    \midrule
    rs7603133 & 2     & \num{3103708} & \num{3.23E-08} & A/G     & \num{0.09} \\
    rs7855681 & 9     & \num{18855498} & \num{3.46E-08} & A/C     & \num{0.32} \\
    \bottomrule
    \end{tabular}%
  \label{tab:sig-FD}%
\end{table}%
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.8\textwidth]{Chapter6/Figures/locuszoom.png}
	\caption[\textbf{Significantly associated loci of left ventricular trabeculation GWAS in genomic context. }]{\textbf{Significantly associated loci of left ventricular trabeculation GWAS in genomic context. }The p-values and genomic location of the two loci reaching genome-wide significance are shown in relation to the p-values of surrounding genotypic markers. Markers are colored by the level of LD they share with the SNP of interest. For both loci, all SNPs that are significantly associated were imputed. For the locus on chromosome~9 (lower panel), an additional SNP which was directly genotypes but has not passed the genome-wide significant level has been marked in red. Generated with LocusZoom \citep{Pruim2010}.}  
	 	\label{fig:locuszoom-fd}
\end{figure}
%
Punctin is a secreted glycoprotein that can be detected in contacts between cells and components of the extra-cellular matrix, but that has not been observed in cell-cell contacts \citep{Hirohata2002}. It is part of the ADAMTS-like protein family which lack the proteolytic activity of their name-lending metalloprotease protein family. While other proteins of the ADAMTS-like family have been shown to be associated with connective tissue disorders and affecting the formation of the extra-cellular matrix \citep{Ahram2009,Hubmacher2015}, the function of punctin remains unknown. However, progress has been made in understanding the regulation of its secretion through post-translational modification of its tryptophane \num{42} residue \citep{Wang2009}. A recently pubished study shows a strong systemic phenotype for the mutation of this tryptophane residue, inhibiting the secretion of the protein. However, no further advances in understanding the mechanisms or finding binding partners of ADAMTSL1 could be made \citep{Hendee2017}.

The locus on chromosome~1 (significant SNP: rs113719231 ) discovered in \cref{section:GWAS-3Dheart} is located near the PRDM16 which has been associated with LVNC \citep{Arndt2013}. A linear model with the rs113719231 genotypes, sex, age, height and weight as explanatory variables and \(\text{FD}_\text{max}^\text{basal}\)/\(\text{FD}_\text{max}^\text{apical}\) as response variables did not show any significant association, even without the burden of the genome-wide significance threshold (\(p = 0.78\)/\(p = 0.77\)).

The clinical phenotype of left ventricular non-compaction has been found associated a number of other cardiac and cardiovascular phenotypes such as conduction abnormalities \citep{Yousef2009}, arrhythmias \citep{Ritter1997,Oechslin2000,Yousef2009}, coronary artery disease \citep{Ritter1997,Junga1999,Jenni2002,Soler2002} and myocardial infarction \citep{Swinkels2007,Toufan2012,Guvens2012}. In addition, a study on population variation of left ventricular trabeculation found associations between the increase in left ventricular trabeculation and prevalence of hypertension, left ventricular mass and wall thickness \citep{Captur2015}. For the majority of these phenotypes original GWAS and meta-analysis of GWAS have been conducted including atrial fibrillation \citep{Gudbjartsson2007,Christophersen2017}, atrioventricular conduction \citep{Denny2010}, coronary heart disease \citep{Schunkert2011,Lee2013a,Nikpay2015}, myocardial infarction \citep{Kathiresan2009,Hirokawa2015,Nikpay2015,Dehghan2016} and blood pressure phenotypes \citep{Ehret2011,Wain2011}. For studies, where the summary statistics of the genome-wide associations were made publicly available (blood pressure phenotypes \citep{Ehret2011,Wain2011}, coronary artery disease \citep{Schunkert2011} and myocardial infarction \citep{Nikpay2015}), I collected the effect size estimate (continuous traits) and odds ratios (case-control setting) for the significantly associated loci on chromosome 2 and 9. The SNP with the highest association on chromosome 9 (rs7855681) was contained in all available studies. For the locus on chromosome 2, the SNP with the highest association was not contained in any of the studies, however rs6758505 which is in strong LD with the discovered SNP in Caucasians (\(r^2=1\)) was found in one of the studies. \Cref{fig:consortia} depicts the effect size estimates and odds ratios for both SNPs estimated for different blood pressure measurements, coronary artery disease and myocardial infarction. For all phenotypes, the confidence intervals of effect size/odds ratio estimates contain zero and one, respectively and thus show no significant effect of the SNPs on these phenotypes. 
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.8\textwidth]{Chapter6/Figures/forestplotCardioConsortia.pdf}
	\caption[\textbf{Effect estimates of significantly associated FD SNPs with other cardiovascular phenotypes. }]{\textbf{Effect estimates of significantly associated FD SNPs with other cardiovascular phenotypes. }Effect size estimates and odds rations for the SNPs significantly associated with FD were derived from previous published studies on blood pressure (BP) phenotypes and risk for coronary artery diseases and myocardial infarction. The diamond indicates the effect estimates, the error bars their confidence interval. The size of the diamond represents the sample size of the study and is normalised to the largest study size (pulse pressure: \(N=\)\num{71663}). All studies were conducted as meta-analyses in the scope of large consortia (faceting labels). The dashed vertical line indicates the value of no effect. }
%For the locus on chromsome 2, a SNP (rs6758505) in perfect lD with the discovered SNP (rs7603133) was contained in the analysis and its values are depicted (). For the locus on chromosome 9 the original SNP was contained in the studies. }  
	 	\label{fig:consortia}
\end{figure}
%
A database search of the GWAS catalogue \citep{MacArthur2017} for significantly associated SNPs and SNPs in LD  (based on entries in the GWAS catalogue, 0.7.08.2018) and the Global Biobank engine \citep{GBE2017} did not yield associations with any other phenotype. 
