\section{Introduction}

\subsection{Outline}
This report resembles a proposed outline for the results chapters of my thesis. I also included a section of the \emph{Introduction} of my thesis about linear models in order to be able to refer to relevant equations. The proposed results chapters describe three major projects I have worked on. In section~\ref{section:limmbo}, I analysed different linear model and linear mixed model set-ups for their validity and calibration in genotype to phenotype mapping depending on population structure, relatedness and heritability. I built on an existing LMM framework to allow for multi-variate modeling of a large number of phenotypes and show the validity of the approach on simulated data and a publically available dataset. In Sections~\ref{section:GWAS_FD} and ~\ref{section:GWAS_pheno3D}, I apply multi-variate models to quantitative traits derived from cardiac magnetic resonance imaging. Two types of heart phenotype data were analysed: i) a low dimensional, trabeculation dataset that describes the lining of the inner layer of the heart muscle and ii) a high dimensional dataset that yields information about heart structure on more the 27,000 coordinates in the left ventricle of the heart. I conducted genome-wide association studies (GWAS) for both datasets.

\paragraph{Thesis outline}
\begin{enumerate}
	\item Introduction
	\begin{itemize}
		\item Genotype to phenotype mapping, linear models, GWAS, ...
		\item Heart development, clinical heart phenotypes, imaging genetics
	\end{itemize}
	\item Results
	\begin{itemize}
		\item Linear Mixed Model Bootstrapping
		\item GWAS of left ventricular trabeculation
		\item GWAS of left ventricular wall thickness
	\end{itemize}
	\item Conclusion
\end{enumerate}

\paragraph{Publication strategy}
We plan to publish three papers, each based on one results chapter. Section~\ref{section:limmbo} was conducted in a collaboration with Oliver Stegle's group, sections~\ref{section:GWAS_FD} and ~\ref{section:GWAS_pheno3D} with Stuart Cook's group at Imperial College London. 

\newpage
\paragraph{Timeline} Proposed timeline for finishing analyses and writing thesis and publications.
\begin{table}[h!]
  \centering
    \begin{tabular}{p{7cm}p{8cm}}
    	\toprule
    	January 2017  & finishing introduction chapter about heart development and clinical heart phenotypes\\
    	January 2017 - February 2017 & finishing main analyses\\
    	February 2017  & finishing introduction chapter about genotype-phenotype mapping\\
     	March 2017 - June 2017 & writing results chapters and papers\\
    	\bottomrule
    \end{tabular}%
  \label{tab:genoOverview}%
\end{table}%



\subsection{Linear models}
In genotype to phenotype mapping, the most simple linear model (LM) describes the linear relationship between a phenotype \(y\) and a genetic marker \(x\). Optionally, known covariates \(F\) can be included as explanatory variables. There are a variety of different types of models, depending on the structure of background effects and the number of phenotypes that are modeled simultaneously. In the following, a short description of the models relevant for this project is outlined. 

\label{sec:ssection:lm}
\paragraph{Uni-variate linear models for genetic analyses}
The uni-variate LM models a single phenotype \(y\) as the sum of a fixed effect of the genetic marker \(x\) and \(K\) known covariates \(F\) and residual noise \(\psi\) across \(N\) samples:

\begin{equation}
\mathbf{y} = \mathbf{F}\boldsymbol{\alpha} + \mathbf{x}\beta + \boldsymbol{\psi},\text{ }
\boldsymbol{\psi}\sim\mathcal{N}\left(\mathbf{0},\sigma_e^2\mathbf{I}_N\right)
\label{eq:lm-uv}
\end{equation}

with
\begin{align*} 
& \text{the phenotype vector } \mathbf{y} \in \mathcal{R}^{N,1},\\
& \text{the matrix of $K$ covariates } \mathbf{F} \in \mathcal{R}^{N,K},\\
& \text{the effect of covariates } \boldsymbol{\alpha} \in \mathcal{R}^{K,1},\\
& \text{the genetic profile of the SNP being tested } \mathbf{x} \in \mathcal{R}^{N,1} \text{and}\\
& \text{the effect size of the SNP } \boldsymbol{\beta} \in \mathcal{R}\\
\end{align*} 


\noindent The association between phenotypes and the genetic markers can be assesed by testing the hypothesis that the genetic variant has an effect \(\beta \neq 0\) versus having no effect on the phenotype. The log likelihood ratio (LLR) test statistic \(\Lambda\) is a commonly used statistic to compare the likelihood of the full model \(H_A\) (Equation~\ref{eq:lm-uv}) with \(\beta \neq 0\) to the one of the Null model \(H_0\) :
\begin{equation}
H_0: \mathbf{y} = \mathbf{F}\boldsymbol{\alpha}  + \boldsymbol{\psi},\text{ }
\boldsymbol{\psi}\sim\mathcal{N}\left(\mathbf{0},\sigma_e^2\mathbf{I}_N\right)
\label{eq:lm_null}
\end{equation}

\noindent The LLR test statistic \(\Lambda\) is defined as
\begin{equation}
\Lambda  =  \mathcal{L} (\hat{\beta}, \hat{\alpha}, \hat{\sigma_{e}}) -  \mathcal{L} (0, \hat{\alpha}, \hat{\sigma_{e}})
\label{eq:llr}
\end{equation}

\noindent where \(\mathcal{L} (\hat{\beta}, \hat{\alpha}, \hat{\sigma_{e}})\) are the maximum likelihood estimators (MLE) of \(H_A\) and \(\mathcal{L} (0, \hat{\alpha}, \hat{\sigma_{e}})\) the MLE of \(H_0\). 

\noindent \(2\Lambda\) follows a \(\chi^2_{df}\) distribution with \(df\) degrees of freedom \citep{Wilks1938} 
\begin{equation}
2\Lambda \sim \chi^2_{df} 
\label{eq:lambda}
\end{equation}

\noindent and allows for the calculation of the P value as :
\begin{equation}
P(\Lambda) = 1 - F_{\chi^2}(2\Lambda, df)
\label{eq:pvalue}
\end{equation}

\paragraph{Multi-variate linear models for genetic analyses.} Extending the model to a multi-variate linear model, i.e. jointly modeling multiple phenotypes \(P\), requires the introduction of trait-design matrices for the fixed effects (\(\mathbf{A}\) and \(\mathbf{B}\) for the covariate and genetic effect respectively) and a trait-by-trait covariance matrix \(\mathbf{C_n}\) for the residual noise:
\begin{equation}
\mathbf{Y} = \mathbf{F}\mathbf{A}\mathbf{W_\alpha} + \mathbf{x}\mathbf{B}\mathbf{W_\beta} + \boldsymbol{\psi},\text{ }
\boldsymbol{\psi}\sim\mathcal{N}\left(\mathbf{C_n} \otimes \mathbf{I}_N\right)
\label{eq:lm-mv}
\end{equation}

with
\begin{align*} 
& \text{the Kronecker product } \otimes \\
& \text{the phenotype matrix } \mathbf{Y} \in \mathcal{R}^{N,P},\\
& \text{the matrix of $K$ covariates } \mathbf{F} \in \mathcal{R}^{N,K},\\
& \text{the effect of covariates } \boldsymbol{A} \in \mathcal{R}^{K,M},\\
& \text{the trait design matrix of the covariates } \mathbf{W_\alpha} \in \mathcal{R}^{M,P},\\
& \text{the genetype vector of the SNP being tested } \mathbf{x} \in \mathcal{R}^{N,1}\\
& \text{the effect size of the SNP } \boldsymbol{B} \in \mathcal{R}^{1, L} \text{and}\\
& \text{the trait design matrix of the genotype } \mathbf{W_\beta} \in \mathcal{R}^{L,P},\\
\end{align*} 

The trait design matrices \(W_\alpha\) and \(W_\beta\) allow different scenarios of the cross-trait architecture of the independent effects on the phenotype.  For instance, in an 'common effect' setup where the genetic variant is assumed to have the same effect across all traits,  \( \boldsymbol{B}\) is equal to \(1_{1xP}\) (\(L=1\)). Allowing different effects across all traits corresponds to  \( \boldsymbol{B} =  \boldsymbol{I}_{P} \) (\(L=P\)). In such a `any effect' setup, the multi-trait modelling simply serves to increase power for detecting genetic variants. 

\subsection{Linear mixed models}
Linear mixed models (LMMs) include both fixed and random effects in the model. In genetics, LMMs can model both fixed genetics effects, i.e. single variants, and background genetic effects, i.e. controling for population structure and accounting for polygenic background \citep{Yu2006}. Population structure and relatedness between individuals can be captured in a genetic relatedness matrix, which accounts for the pairwise genetic similarity between individuals. The relatedness matrix is estimated as
 \begin{equation}
 R = \frac{1}{S}XX^T
 \label{eq:relatedness}
 \end{equation}
 where \(S\) is the number of SNPs used for the estimation and \(X\) is the \(S \times N\) genotype matrix. As described for the linear model, there are uni-variate and multi-variate LMM set-ups for genetic association analyses. 


\paragraph{Uni-variate LMM for genetic analyses.} The genetic background and residual noise are modeled as random effects with a scalar MLE for the genetic \(g\) and noise trait-variance \(\psi\),   \(\sigma_g^2\) and \(\sigma_e^2\):

\begin{equation}
\mathbf{y} = \mathbf{F}\boldsymbol{\alpha} + \mathbf{x}\beta + \mathbf{g}+\boldsymbol{\psi},\text{ }
\mathbf{g}\sim\mathcal{N}\left(\mathbf{0},\sigma_g^2\mathbf{R}\right),\text{ }
\mathbf{\psi}\sim\mathcal{N}\left(\mathbf{0},\sigma_e^2\mathbf{I}_N\right)
\label{eq:lmm-uv}
\end{equation}

with
\begin{align*} 
& \text{the phenotype vector } \mathbf{y} \in \mathcal{R}^{N,1},\\
& \text{the matrix of $K$ covariates } \mathbf{F} \in \mathcal{R}^{N,K},\\
& \text{the effect of covariates } \boldsymbol{\alpha} \in \mathcal{R}^{K,1},\\
& \text{the genetic profile of the SNP being tested } \mathbf{x} \in \mathcal{R}^{N,1},\\
& \text{the effect size of the SNP } \boldsymbol{\beta} \in \mathcal{R} \text{ and}\\
& \text{the sample relatedeness matrix } \mathbf{R} \in \mathcal{R}^{N,N},
\end{align*} 

\paragraph{Multi-variate LMM for genetic analyses.} In the multi-variate case, the MLE of the trait covariances are \(P\times P\) trait-by-trait covariance matrices \(C_g\) and \(C_n\) for the genetic and noise components, respectively (Equation \ref{eq:lmm-mv}).

\begin{equation}
\mathbf{Y} = \mathbf{F}\mathbf{A}\mathbf{W_\alpha} + \mathbf{x}\mathbf{B}\mathbf{W_\beta} + \mathbf{g}+\boldsymbol{\psi},\text{ }
\boldsymbol{g}\sim\mathcal{N}\left(\mathbf{C_g} \otimes \mathbf{R}_N\right),\text{ }
\boldsymbol{\psi}\sim\mathcal{N}\left(\mathbf{C_n} \otimes \mathbf{I}_N\right)
\label{eq:lmm-mv}
\end{equation}

with
\begin{align*} 
& \text{the Kronecker product } \otimes \\
& \text{the phenotype matrix } \mathbf{Y} \in \mathcal{R}^{N,P},\\
& \text{the matrix of $K$ covariates } \mathbf{F} \in \mathcal{R}^{N,K},\\
& \text{the effect of covariates } \boldsymbol{A} \in \mathcal{R}^{K,M},\\
& \text{the trait design matrix of the covariates } \mathbf{W_\alpha} \in \mathcal{R}^{M,P},\\
& \text{the genetype vector of the SNP being tested } \mathbf{x} \in \mathcal{R}^{N,1}\\
& \text{the effect size of the SNP } \boldsymbol{B} \in \mathcal{R}^{1, L} \text{and}\\
& \text{the trait design matrix of the genotype } \mathbf{W_\beta} \in \mathcal{R}^{L,P},\\
\end{align*} 


