\section{Multi-trait GWAS detects three loci associated with heart wall thickness}
\label{section:GWAS-3Dheart}
Treating the \num{19} components as proxies for the true phenotypes, I was then able to conduct a mtGWAS to capture the genetics of left ventricular wall thickness. Based on previous studies \citep{Price2006,Patterson2006} and results obtained in \cref{section:calibration-limmbo,fig:calibration-LM}, we know that mtGWAS in cohorts with little population structure and no relatedness is well calibrated. In order to avoid confounding relationship structure in the dimensionality reduction step, I had removed related individuals and individuals that were not of Caucasian ancestry (\cref{subsection:genotypes}). Given this genotype structure, I used a simple linear model with components as response and genotypes as explanatory variables for the mtGWAS of the low-dimensional heart phenotypes. As there are no prior assumptions about the genotype effects, I modeled the SNP effects based on an any effect design matrix (\cref{section:model-design}). 

The results of the mtGWAS are depicted in \cref{fig:manhattan-heart}, with three loci that pass the genome-wide significance level. The qq-plot in \cref{fig:qq-heart} shows a well-calibrated behavior.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/lm_mt_pcs_manhattanplot.png}
	\caption[\textbf{Manhattan plot of the multi-trait GWAS on 3D heart phenotypes .}]{\textbf{Manhattan plot of the multi-trait GWAS on 3D heart phenotypes. }The \num{19} stable components derived from PCA, Isomap and LaplacianEigenmap were modeled jointly in an any effect multi-trait GWAS. The p-values of all genome-wide SNPs are depicted. The horizontal grey line is drawn at the level of genome-wide significance: \(p = 5 \times 10^{-8}\). Two loci on chromosome~1 and one locus on chromosome~10 pass the genome-wide significance level.} 
	 	\label{fig:manhattan-heart}
\end{figure}

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.5\textwidth]{Chapter5/Figures/lm_mt_pcs_qqplot.png}
	\caption[\textbf{Quantile-quantile plot of the multi-trait GWAS on 3D heart phenotypes .}]{\textbf{Quantile-quantile plot of the multi-trait GWAS on 3D heart phenotypes. } The observed genome-wide p-values are plotted against p-values drawn from a uniform distribution in \([0,1]\) of the same sample size (expected p-values). The diagonal line starts at the origin and has slope one.} 
	 	\label{fig:qq-heart}
\end{figure}

\Cref{tab:gwas-heart} summarises the chromosomal location, p-values and SNP information of the most strongly associated SNPs per locus. Their genomic context is displayed in \cref{fig:locuszoom-heart}. The locus with the strongest association is located in a regulatory region of a gene-rich area between the SKI gene on the forward and the MORN1 gene on the reverse strand (\cref{fig:locuszoom-heart}, upper panel; \cref{fig:regulatory-heart}). SKI is developmental gene where \textit{de novo} mutations are associated with a complex early developmental syndrome (Shprintzen-Goldberg syndrome) with cranofacial, bone development and cardiovascular phenotypes \citep{Greally2013}. Zebrafish knockdown models of SKI orthologs give rise to complex developmental changes, including cardiac phenotypes \citep{Doyle2012}. In addition, a non-developmental phenotype for altered expression of a SKI orthologues was observed in rat cardiomyocytes. In this system, the overexpression of the rat SKI orthologue leads to a decrease in fibroblast-to-myofibroblast phenoconversion, the main mechanisms for fibrotic heart disease \citep{Cunnington2010,Cunnington2014,Zeglinski2016}. Taken together, these studies show an involvement of SKI genes in a variety of cardiac phenotypes across different tissues stages. The other gene in proximity of rs139971383, the MORN1 gene, is relatively unstudied. 

The second locus on chromosome \num{1} lies within intron nine of the MEGF6 gene (\cref{fig:locuszoom-heart}, middle panel), which encodes for a secreted, calcium-iron binding protein \citep{Nakayama1998}. It is also in proximity of the PRDM16 gene, wherein deletions and mutations were shown to be implicated in two types of cardiomyopathies, left ventricular non-compaction (\cref{section:intro-FD}) and dilated cardiomyopathy  (\cref{subsection:CVD}) \citep{Arndt2013}. Based on zebrafish models of the observed human genotypes, the authors propose that PRDM16 mutations lead to a decreased proliferative capacity during cardiogenesis. Interestingly, the study also found a link between the SKI and PRDM16 genes, suggesting a functional synergy that leads to descreased cardiac outpout in zebrafish models with knock-down phenotypes of SKI and PRDM16. rs143266802 is located downstream of the zink finger protein-encoding gene ZNF487 (\cref{fig:locuszoom-heart}, lower panel).  

A database search of the GWAS catalogue \citep{MacArthur2017} (based on entries in the GWAS catalogue, 0.7.08.2018) and the Global Biobank engine, a resource for estimated genetic effects on cancers, autoimmune diseases, psychiatric, neurological, and cardiometabolic diseases \citep{BiobankEngine2017} did not yield any results. 


% Table generated by Excel2LaTeX from sheet 'Association3Dheart'
\begin{table}[htbp]
  \centering
  \caption[\textbf{Strongest genotype-phenotype association per locus for 3D heart GWAS. }]{\textbf{Strongest genotype-phenotype association per locus for 3D heart GWAS. } For each significant locus, the p-values for SNPs in LD with an \(r^2 > 0.8\) in a \num{50}kb window were compared and only the SNP with smallest p-value per locus listed below. MAF: minor allele frequency. }
    \begin{tabular}{lllllll}
    \toprule
    SNP   & Chr   & Position & P-value & Minor allele & Major allele & MAF \\
    \midrule
    rs139971383 & \num{1} & \num{2246921} & \num{1.09E-10} & C     & G     & \num{0.013} \\
    rs113719231 & \num{1} & \num{3427138} & \num{9.04E-09} & C     & T     & \num{0.11} \\
    rs143266802 & \num{10} & \num{43978849} & \num{1.54E-08} & C     & T     & \num{0.022} \\
    \bottomrule
    \end{tabular}%
  \label{tab:gwas-heart}%
\end{table}%


\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.65\textwidth]{Chapter5/Figures/locuszoom.png}
	\caption[\textbf{Significantly associated loci of the 3D heart GWAS in genomic context. }Generated with LocusZoom \citep{Pruim2010}]{\textbf{Significantly associated loci of the 3D heart GWAS in genomic context. }The p-values and genomic location of the \num{3} significantly associated loci from the mtGWAS on the stable components from PCA, LaplacianEigenmaps and Isomap are shown in relation to the p-values of surrounding genotypic markers. Markers are colored by the level of LD they share with the SNP of interest. There was no LD information available on LocusZoom for the locus depicted in the bottom panel.} 
	 	\label{fig:locuszoom-heart}
\end{figure}

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/Human_121595772323196.pdf}
	\caption[\textbf{Regulatory context of locus with strongest association. }]{\textbf{Regulatory context of locus with strongest association. } The SNP with the strongest association (rs139971383) in the mtGWAS lies in a promoter flanking region epigentically active in myocytes from the left ventricle (Ensembl, Human Regulatory Features, GRCh37.p13). } 
	 	\label{fig:regulatory-heart}
\end{figure}

The multi-trait LM per SNP yields individual effect size estimates for each trait that is jointly modeled. There are two ways by which these effect size estimates can be helpful in understanding the genotype-phenotype association. First, traits driving the assocation with the SNP are expected to have high effect size estimates. Second, traits that are similarily affected by the SNPs will have similar effect size estimates.  In \cref{fig:effectsizes-heart}, I show the effect sizes for each of the \num{19} components per SNP clustered by their Euclidean distance. For the locus most significantly associated with the \num{19} proxy phenotypes of wall thickness, there are two clusters of high effect size estimates (\cref{fig:effectsizes-heart}, rs139971383). While one of them contains components from one method only (LaplacianEigenmaps1, 2, and 6), the other cluster contains two components from different methods, Isomap1 and PCA1. Similarily, the association of rs143266802 seems to be driven by a combination of components from all three methods (PCA2, Isomap2 and LaplacianEigenmap3). These results demonstrate the strength of this analysis approach, where different aspects of phenotype morphology are captured by different methods that can then jointly represent a wider aspect of the phenotype structure. In contrast, independent analysis of components from a single method, could not detect these strong signals (\cref{fig:manhattan-3Dheart-single}). Only the locus situated in the regulatory region between MORN1 and SKI was detected in a mtGWAS with the components from Laplacian Eigenmaps alone (\cref{fig:manhattan-3Dheart-single}\subfig{A}); p-value: \num{1.36E-08}), confirming the effect size cluster structure observed for this locus, with large effect size for LaplacianEigenmaps1, 2, and 6. Additional signal for this independent analysis was overall weaker than the one for the combined analyses. GWAS with components from Isomap and PCA alone did not yield any significant associations (\cref{fig:manhattan-3Dheart-single}\subfig{B} and \subfig{C}). 

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/effectsizes.pdf}
	\caption[\textbf{Clustering of effect size estimates from the 3D heart GWAS. }]{\textbf{Clustering of effect size estimates from the 3D heart GWAS. } The effect size estimates from the most strongly associated SNPs at each locus were clustered across components and SNPs by average-linkage hierarchical clustering of their Euclidean distances. The dendrogram of the components is labeled based on the methods used to generate the low-dimensional representation. The numbering indicates the position of the component as returned from the algorithm, i.e. for PCA the ordering based on the amount of variance explained. LE: Laplacian Eigenmaps; IM: Isomap.} 
	 	\label{fig:effectsizes-heart}
\end{figure}

The proxy phenotypes are critical for the discovery of the genetic association. In order to understand the effect on the underlying biology without mediation via the dimensionality reduction methods, I linked the SNPs back to the original heart phenotypes.

In a first, simple approach, I used the discovered SNPs as explanatory variables in a simple linear model with left ventricular mass as the phenotype and sex, age, height and weight as additional covariates. None of the three SNP discovered with the multi-trait GWAS shows significant association with left ventricular mass (rs139971383: \(p=0.89\) , rs11371923: \(p=0.22\) , rs143266802: \(p=0.68\)). This results is not discouraging, however, since the hypothesis was that stable components capture regional variation in left ventricular wall thickness. Summarising wall thickness variation in a single scalar values such as left ventricular mass might not be able to capture these regional changes in mass. In order to analyse if the discovered SNPs show association specific to certain regions in the left ventricle, I evaluated the relationship between the genotypes of the strongest associated SNP and the original, spatially-resolved left ventricular wall thickness measurements. 
For each of the \num{27623} positions, I conducted a simple linear model with covariate-adjusted wall thickness measurements (data identical to input data for dimensionality reduction, \cref{section:DimRed-heart}) as the response variable and the genotype of rs139971383 as the explanatory variable. \Cref{fig:wall-heart} shows these associations with the SNP in relation to their location on the left ventricle. Importantly, although none of these associations would be likely to survive the large multiple testing burden if used for discovery, they do show a specific location of the left ventricle which is affected by this SNP.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.8\textwidth]{Chapter5/Figures/rs139971383onHeart.pdf}
	\caption[\textbf{Association of  rs139971383 with left ventricular wall thickness. } Plot generated with ParaView.]{\textbf{Association of rs139971383 with left ventricular wall thickness. }The \num{27623} covariate-adjusted wall thickness measurements in the left ventricle were used as the response variable in a simple linear model with the genotype of rs139971383 as the explanatory variable. The -\(log10(\text{p-value})\) of the association of each models is projected onto its corresponding 3D position. Darker colors indicate stronger associations. } 
	 	\label{fig:wall-heart}
\end{figure}


