\section{Multi-trait GWAS detects three loci associated with heart wall thickness}
\label{section:GWAS-3Dheart}
Treating the \num{19} components as proxies for the true phenotypes, I was then able to conduct a \gls{mtgwas} to capture the genetics of left ventricular wall thickness. Based on previous studies \citep{Price2006,Patterson2006} and results obtained in \cref{section:calibration-limmbo,fig:calibration-LM}, we know that \gls{mtgwas} is well calibrated in cohorts with little population structure and no relatedness. In order to avoid confounding relationship structure in the dimensionality reduction step, I had already removed related individuals and individuals that were not of European ancestry (\cref{subsection:genotypes}). Given this genotype structure, I used a simple linear model with components as response and genotypes as explanatory variables for the \gls{mtgwas} of the low-dimensional heart phenotypes. As there are no prior assumptions about the genotype effects, I modelled the \gls{snp} effects based on an any effect design matrix (\cref{subsubsection:model-design}). 

The results of the \gls{mtgwas} are depicted in \cref{fig:manhattan-heart}, with three loci that pass the genome-wide significance level of \(5 \times 10^{-8}\). The qq-plot in \cref{fig:qq-heart} shows a well-calibrated test statistic.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/lm_mt_pcs_manhattanplot.png}
	\caption[\textbf{Manhattan plot of the multi-trait GWAS on 3D heart phenotypes .}]{\textbf{Manhattan plot of the multi-trait GWAS on 3D heart phenotypes. }The \num{19} stable components derived from \gls{pca}, Isomap and Laplacian Eigenmap were modelled jointly in an any effect \gls{mtgwas}. The p-values of all genome-wide \glspl{snp} are depicted. The horizontal grey line is drawn at the level of genome-wide significance: \(p = 5 \times 10^{-8}\). Two loci on chromosome~1 and one locus on chromosome~10 pass the genome-wide significance level.} 
	 	\label{fig:manhattan-heart}
\end{figure}
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.5\textwidth]{Chapter5/Figures/lm_mt_pcs_qqplot.png}
	\caption[\textbf{Quantile-quantile plot of the multi-trait GWAS on 3D heart phenotypes .}]{\textbf{Quantile-quantile plot of the multi-trait GWAS on 3D heart phenotypes. } The observed genome-wide p-values are plotted against p-values drawn from a uniform distribution in \([0,1]\) of the same sample size (expected p-values). The diagonal line starts at the origin and has slope one.} 
	 	\label{fig:qq-heart}
\end{figure}
%
\Cref{tab:gwas-heart} summarises the chromosomal location, p-values and \gls{snp} information of the most strongly associated \glspl{snp} per locus. Their genomic context is displayed in \cref{fig:locuszoom-heart}. The locus with the strongest association is located in a regulatory region of a gene-rich area between the \textit{SKI} gene on the forward and the \textit{MORN1} gene on the reverse strand (\cref{fig:locuszoom-heart}, upper panel; \cref{fig:regulatory-heart}). \textit{SKI} is developmental gene where \textit{de novo} mutations are associated with a complex early developmental syndrome (Shprintzen-Goldberg syndrome) with cranofacial, bone development and cardiovascular phenotypes \citep{Greally1993}. Zebrafish knockdown models of \textit{SKI} orthologs give rise to complex developmental changes, including cardiac phenotypes \citep{Doyle2012}. In addition, a non-developmental phenotype for altered expression of a \textit{SKI} orthologues was observed in rat cardiomyocytes. In this system, the overexpression of the rat \textit{SKI} orthologue leads to a decrease in fibroblast-to-myofibroblast phenoconversion, the main mechanisms for fibrotic heart disease \citep{Cunnington2010,Cunnington2014,Zeglinski2016}. Taken together, these studies show an involvement of \textit{SKI} genes in a variety of cardiac phenotypes across different tissues stages. The other gene in proximity to rs139971383, the \textit{MORN1} gene, is relatively unstudied. 

The second locus on chromosome \num{1} lies within intron nine of the \textit{MEGF6} gene (\cref{fig:locuszoom-heart}, middle panel), which encodes for a secreted, calcium-iron binding protein \citep{Nakayama1998}. It is also in proximity to the \textit{PRDM16} gene, wherein deletions and mutations were shown to be implicated in two types of cardiomyopathies, left ventricular non-compaction (\cref{section:intro-FD}) and dilated cardiomyopathy  (\cref{subsection:CVD}) \citep{Arndt2013}. Based on zebrafish models of the observed human genotypes, the authors propose that \textit{PRDM16} mutations lead to a decreased proliferative capacity during cardiogenesis. Interestingly, the study also found a link between the \textit{SKI} and \textit{PRDM16} genes, suggesting a functional synergy that leads to decreased cardiac output in zebrafish models with knock-down phenotypes of \textit{SKI} and \textit{PRDM16}. rs143266802 is located downstream of the zinc finger protein-encoding gene \textit{ZNF487} (\cref{fig:locuszoom-heart}, lower panel), which has no associated phenotypes in human (GRCh38.p10, ensembl release 90, \citep{Aken2016}).  

A database search of the \gls{gwas} catalogue \citep{MacArthur2017} (based on entries in the \gls{gwas} catalogue, 0.7.08.2018) and the Global Biobank engine, a resource for estimated genetic effects on cancers, autoimmune diseases, psychiatric, neurological, and cardiometabolic diseases \citep{GBE2017} did not yield any other phenotypes that these \glspl{snp} were associated with.


% Table generated by Excel2LaTeX from sheet 'Association3Dheart'
\begin{table}[htbp]
  \centering
  \caption[\textbf{Strongest genotype-phenotype association per locus for 3D heart GWAS. }]{\textbf{Strongest genotype-phenotype association per locus for 3D heart GWAS. } For each locus, the p-values for \glspl{snp} in \gls{ld} with an \(r^2 > 0.8\) in a \num{50}kb window were compared and only the \gls{snp}  with smallest p-value per locus listed below. Gene: gene in proximity to \gls{snp}  and described in detail in the text above. M: major allele, m:  minor allele, MAF: minor allele frequency. }
  \begin{small}
    \begin{tabular}{lllllll}
    \toprule
    SNP   & Gene & Chr   & Position & P-value &  M/m allele & MAF \\
    \midrule
    rs139971383 & \textit{SKI} & \num{1} & \num{2246921} & \num{1.09E-10} & C/G     & \num{0.013} \\
    rs113719231 & \textit{PRDM16} & \num{1} & \num{3427138} & \num{9.04E-09} & C/T     & \num{0.11} \\
    rs143266802 & \textit{ZNF487} & \num{10} & \num{43978849} & \num{1.54E-08} & C/T     & \num{0.022} \\
    \bottomrule
    \end{tabular}%
    \end{small}
  \label{tab:gwas-heart}%
\end{table}%
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.65\textwidth]{Chapter5/Figures/locuszoom.png}
	\caption[\textbf{Genomic context of loci associated loci with 3D heart phenotypes. }]{\textbf{Genomic context of loci associated loci with 3D heart phenotypes . }The p-values and genomic location of the three associated loci from the \gls{mtgwas} on the stable components from \gls{pca}, Laplacian Eigenmaps and Isomap are shown in relation to the p-values of surrounding genotypic markers. Markers are coloured by the level of \gls{ld} they share with the \gls{snp}  of interest. There was no \gls{ld} information available on LocusZoom for the locus depicted in the bottom panel. Generated with LocusZoom \citep{Pruim2010}.}  
	 	\label{fig:locuszoom-heart}
\end{figure}
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/Human_121595772323196.pdf}
	\caption[\textbf{Regulatory context of locus with strongest association. }]{\textbf{Regulatory context of locus with strongest association. } The \gls{snp} with the strongest association (rs139971383) in the \gls{mtgwas} lies in a promoter flanking region epigenetically active in myocytes from the left ventricle (Ensembl, Human Regulatory Features, GRCh37.p13). } 
	 	\label{fig:regulatory-heart}
\end{figure}
%
The \gls{mvlm} per \gls{snp} yields individual effect size estimates for each trait that is jointly modelled. There are two ways by which these effect size estimates can be helpful in understanding the genotype-phenotype association. Firstly, traits driving the association with the \gls{snp} are expected to have high effect size estimates. Secondly, traits that are similarly affected by the \glspl{snp} will have similar effect size estimates.  In \cref{fig:effectsizes-heart}, I show the effect sizes for each of the \num{19} components per \gls{snp} clustered by their Euclidean distance. For the locus with the strongest associated with the \num{19} proxy phenotypes of wall thickness, there are two clusters of high effect size estimates (\cref{fig:effectsizes-heart}\subfig{A}, rs139971383). While one of them contains components from one method only (LaplacianEigenmaps1, 2, and 6), the other cluster contains two components from different methods, Isomap1 and PCA1. Similarily, the association of rs143266802 seems to be driven by a combination of components from all three methods (PCA2, Isomap2 and LaplacianEigenmap3). These results demonstrate the strength of this analysis approach, where different aspects of phenotype morphology are captured by different methods that can then jointly represent a wider aspect of the phenotype structure. The corresponding trait correlations are shown in \cref{fig:effectsizes-heart}\subfig{B}. A number of effect size clusters can seemingly be explained by the strong correlation of their respective traits (indicated by coloured boxes).
In contrast, independent analysis of components from a single method, could not detect these strong signals (\cref{fig:manhattan-3Dheart-single}). Only the locus situated in the regulatory region between \textit{MORN}1 and \textit{SKI} was detected in a mtGWAS with the components from Laplacian Eigenmaps alone (\cref{fig:manhattan-3Dheart-single}\subfig{A}); p-value: \num{1.36E-08}), confirming the effect size cluster structure observed for this locus, with large effect size for LaplacianEigenmaps1, 2, and 6. Additional signal for this independent analysis was overall weaker than the one for the combined analyses. \gls{gwas} with components from Isomap and \gls{pca} alone did not yield any associations (\cref{fig:manhattan-3Dheart-single}\subfig{B} and \subfig{C} in the appendix). 
%
\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter5/Figures/effectsizes_correlation.pdf}
	\caption[\textbf{Effect size estimates and trait correlation from the 3D heart GWAS. }]{\textbf{Effect size estimates and trait correlation from the 3D heart GWAS. } A. The effect size estimates from the most strongly associated \glspl{snp} at each locus were clustered across components and \glspl{snp} by average-linkage hierarchical clustering of their Euclidean distances. The dendrogram of the components is labelled based on the methods used to generate the low-dimensional representation. The numbering indicates the position of the component as returned from the algorithm, i.e. for \gls{pca}  the ordering based on the amount of variance explained. LE: Laplacian Eigenmaps; IM: Isomap. B. Trait-trait correlations of the corresponding low-dimensional representations used as response variable in the GWAS. For comparison, the order of the traits was matched to the clustering of the effect sizes. Effect size clustered where strong trait-trait correlations are observed are indicated by colour-matched boxes.} 
	 	\label{fig:effectsizes-heart}
\end{figure}
%
The proxy phenotypes are critical for the discovery of the genetic association but do not necessarily represent a biologically meaningful conformation. In order to understand the effect on the underlying biology without mediation via the dimensionality reduction methods, I linked the \glspl{snp} back to the original heart phenotypes.

In a first, simple approach, I used the discovered \glspl{snp} as explanatory variables in a simple linear model with left ventricular mass as the phenotype and sex, age, height and weight as additional covariates. None of the three \gls{snp} discovered with the \gls{mtgwas} shows association with left ventricular mass (rs139971383: \(p=0.89\) , rs11371923: \(p=0.22\) , rs143266802: \(p=0.68\)). This result is not discouraging, however, since the hypothesis was that stable components capture regional variation in left ventricular wall thickness. Summarising wall thickness variation in a single scalar value such as left ventricular mass might not be able to capture these regional changes in mass. In order to analyse if the discovered \glspl{snp} show association specific to certain regions in the left ventricle, I evaluated the relationship between the genotypes of the strongest associated \gls{snp} and the original, spatially-resolved left ventricular wall thickness measurements. 
For each of the \num{27623} positions, I conducted a simple linear model with covariate-adjusted wall thickness measurements (data identical to input data for dimensionality reduction, \cref{section:DimRed-heart}) as the response variable and the genotype of rs139971383 as the explanatory variable. \Cref{fig:wall-heart} shows these associations with the \gls{snp} in relation to their location on the left ventricle. Importantly, although none of these associations would be likely to survive the large multiple testing burden if used for discovery, they do show a specific localisation to the left ventricle which is affected by this \gls{snp}.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.8\textwidth]{Chapter5/Figures/rs139971383onHeart.pdf}
	\caption[\textbf{Association of  rs139971383 with left ventricular wall thickness. }]{\textbf{Association of rs139971383 with left ventricular wall thickness. }The \num{27623} covariate-adjusted wall thickness measurements in the left ventricle were used as the response variable in a simple linear model with the genotype of rs139971383 as the explanatory variable. The -\(log10(\text{p-value})\) of the association of each models is projected onto its corresponding 3D position. Darker colors indicate stronger associations. Generated with ParaView. } 
	 	\label{fig:wall-heart}
\end{figure}


