\section{LiMMBo}
\label{section:limmbo}
Multi-trait models have been used in quantitative genetics since the mid-1990ies \citep{Jiang1995} and a variety of flavors are available such as methods based on dimension reduction techniques (e.g. principal component analysis or canonical correlation analysis) and multivariate models (reviewed in \citep{Shriner2012}). Korte and colleagues were the first to combine one of these methods with GWAS statistical tools to allow for multi-trait linear-mixed modeling of paired-phenotypes on a genome-wide level \citeyearpar{Korte2012}. Modeling traits jointly can improve the statistical power by accounting for correlated background variation as well as by combining weak genetic effects across traits. Accounting for correlated background allows the observed variance of the phenotype to be decomposed into different variance components such as genetic and environment components \citep{Korte2012}. Most multi-trait linear mixed model (mtLMM) frameworks are optimised for large numbers of samples (N) and SNPs but are limited to a moderate number of phenotypic traits (P, usually less than 10), as complexity can reach up to \(O(P^6)\) \citep{Zhou2014,Casale2015}.  In studies with more than ten phenotypic traits, alternative methods such a meta-analysis of single-trait setups have been applied \citep{Bolormaa2014}. I investigated different linear model set-ups for analysing datasets of more than 30 traits depending on the relatedness and population structure of the test cohort and complexity of the traits. Suitability of different methods was assessed by testing calibration of the models under the null hypothesis of no genetic association on simulated datasets with differing levels of relatedness and population structure. I developed a linear mixed model bootstrapping (LiMMBo)-based method that can analyse datasets with more than 30 phenotypes in a mtLMM framework when applicable. An increase in power compared to uni-variate models is demonstrated on a publically available yeast data set of 41 previously published quantitive traits \citep{Bloom2013}, by comparing the results of single-trait (st) to multi-trait (mt) GWAS. I show both in simulation and in real datasets that this estimation gives similar results for up to 30 phenotypes, but can scale up to 100 phenotypes.


