\chapter{PhenotypeSimulator}
\label{chapter:simulation} 
For method development in quantitative genetics, one often needs a set of well-characterised genotypes and phenotypes to know the ground truth based on which comparisons of the model performance can be made. In the context of this thesis, genotype and phenotype simulations were crucial for the development of a new method for multi-trait mapping of high-dimensional datasets (\cref{chapter:limmbo}) and the evaluation of different dimensionality reduction techniques (\cref{chapter:DimReduction}).

The complexity of the simulated phenotype components will depend on the specifics of the model that is being developed. With the detailed whole-genome genotype data available through standard techniques such as genotyping arrays and subsequent imputation and the measurement of multiple traits per sample, the complexity of the hypotheses for testing the underlying genetics of the observed phenotypes have increased. Models range from simple linear models with a few fixed effects on a single trait to complex linear mixed models with fixed and random effect components on multiple traits \citep{Stephens2013,Marigorta2014,Zhou2014,Loh2014}. With the increase in analysis complexity, sophisticated approaches for modelling realistic genotype and phenotype structures are needed. These simulated genotypes and phenotypes reflect our perceived understanding of the true phenotype structure and do not guarantee the biologically correctness of real phenotypes. However, they are invaluable in model design, as any model showing flawed statistics on the possibly simplified biological model will suffer from at least the same flaws on the true biological data.

In this chapter, I will first describe simulation strategies for genotypes with different levels of population structure and relatedness. Following that, I introduce the phenotype simulation strategy used for all simulated datasets within this thesis. In order to broadly distribute this simulation framework, I have developed \textit{PhenotypeSimulator}, an easily accessible tool for phenotype simulation that allows for a flexible and customisable simulation set-up. \textit{PhenotypeSimulator} can be installed from the Comprehensive R Archive Network \citep{Meyer2017b} and is currently under review for publication \citep{Meyer2017a}.

\section{Genotype simulation}
\label{section:genotype-simulation}
There are a number of different strategies to generate genotype data for genetic association studies. In the most simple case and assuming bi-allelic SNPs, each SNP is simulated from a binomial distribution with two trials and probability equal to the given allele frequencies  (e.g in \citep{Lippert2013}). This simple approach, however, does not simulate any dependency between the genotypes as is observed with LD structure in the genome. In order to mimic genomic LD structure and allele frequency distributions in the simulated dataset, three general approaches exist: i) backward-time or coalescent simulation, ii) forward time and iii) resampling approaches. The coalescent \citep{Hudson2002,Ewing2010,Kelleher2016} and foward-time approaches \citep{Peng2007,Hoggart2007,Carvajal-Rodriguez2008} use population genetic models to simulate genotypes and are particularly useful for studying evolution and demography. However, they often suffer from computational demands for diploid genome-wide SNP data \citep{Liu2008,Yuan2012}. Resampling approaches offer a practical solution that can be used to efficiently generate genetic data with different relatedness and population structures, which is particularly useful in genetic association studies. It was proposed by \citet{Wright2007} and consists in combining existing genotype data into the genotypes of the simulated samples, thereby retaining allele frequency and LD patterns. 

I choose to follow the resampling strategies described in \citep{Loh2014,Casale2015} where the genotypes of each individual are simulated as mosaics of real genotypes from different populations. As the sampling dataset, I used the genotype data from \num{365} individuals of four European ancestry populations (CEU, FIN, GBR, TSI) from the 1000 Genomes (1KG) Project \citep{1000Genomes2015}.  First, each individual is randomly assigned a predefined number of unique original genotypes which will serve as its ancestors. Subsequently, these diploid ancestors' genomes are split into blocks of \num{1000} SNPs. For each SNP block, one of the ancestor is chosen at random and its genotype is copied to the individual's genome. 

The number and the subpopulation of ancestors that are chosen for simulating the genomes of a new cohort are critical for controlling the level of structure within the cohort. The number of ancestors sets the level of relatedness within the cohort. Low numbers of \(N\) introduce relatedness among individuals, while high numbers of \(N\) lead to low levels of structure and relatedness.  For instance, with \(N=2\), each individual in the newly synthesised cohort is composed of genotypes from only two out of the \num{365} individuals. For individuals where a same ancestor was drawn at random (probability \(p=0.005\)), each SNP block has a \num{25}\% probability of being the same between these indivuals. With \(N=10\), the probability for a common ancestor increases (\(p=0.03\)), however the sharing of SNP blocks decreases to \num{1}\%. 

The choice of subpopulation determines the level of population structure in the simulated genotypes: allowing for random selection of ancestors independent from the four subpopulations in the 1KG datasets yields low levels of population structure. Including an \textit{a priori} random selection of subpopulation step before selecting the ancestors for an individual and subsequently restricting ancestor selection to this subpopulation will gives rise to population structure in the simulated cohort. 

I simulated three genotype sets, each with 1,000 samples, that differed i) in the number of ancestors \(N\) from which the genotypes were chosen and ii) the subpopulations the ancestors were chosen from:
\begin{enumerate}[label=\Alph*.]
\item unrelatedPopStructure: unrelated individuals with prior assignment of ancestral population  (\(N=10\), i.e. only CEU or only FIN or only GBR or only TSI)
\item unrelatedNoPopStructure: unrelated individuals with mixed ancestral population  (\(N=10\), i.e. CEU and FIN and GBR and TSI)
\item relatedNoPopStructure: related individuals with mixed ancestral population (\(N=2\), i.e. CEU and FIN and GBR and TSI))
\end{enumerate}


The level of structure and relatedness introduced by this simulation strategy can be visualised by examining the genetic relationship matrix and the principal components of the genotypes. The genetic relationship matrix is estimated as a realised relationship matrix via \cref{eq:relatedness} and serves as a measure for relatedness between the individuals, while principal components reflect the genotypic variance in the data (\cref{subsubsection:relatedness-model-variables}). The hierarchical clustering of the genetic relationship estimates and scatter plots of the first two principal components for each genotype set are shown in \cref{fig:kinship-matrices}. Samples cluster tightly based on their ancestral subpopulations (\cref{fig:kinship-matrices}\subfig{A}), while there is no clustering and an even spread in the PC plot for the cohort of unrelated individuals with ancestors sampled across all subpopulations (\cref{fig:kinship-matrices}\subfig{B}). The cohort of related individuals shows less spread in the second principal component and higher individual genetic relationship  estimates (\cref{fig:kinship-matrices}\subfig{C}).


\begin{figure}[htbp]
	\centering
	\includegraphics[page=2, trim = 0mm 10mm 15mm 0mm, clip, scale=0.25]{Chapter1/Figures/simulatedCovarianceMatrices_kinship.png}
	\caption[\textbf{Genetic relationship matrices and principal components of three simulated European ancestry cohorts.}]{\textbf{Genetic relationship matrices and principal components of three simulated European ancestry cohorts.} The genotypes were simulated based on genotype data from four European ancestry populations (ancestry colour key in panel A). Depending on the choice and number of ancestors for the sampling of chromosomes to simulate an individual's genotype, cohorts with differing levels of population and relatedness structure will arise. The left column depicts the hierarchical clustering of the sample-by-sample  genetic relationship coefficients (complete linkage clustering of Euclidean distance between coefficients), the right column the first and second principal component (PC) of the sample genotypes for the three different cohorts: A. unrelated individuals, with population structure: \(N=10\), prior assignment to ancestral population. B. unrelated individuals, no population structure: \(N=10\), no prior assignment to ancestral population. C. related individuals, no population structure: \(N=2\), no prior assignment to ancestral population.}
 	\label{fig:kinship-matrices}
\end{figure}


\section{Phenotype simulation}
\label{section:phenotype-simulation}

Phenotypes are typically generated as the sum of genetic effects, effects from non-genetic factors and observational noise. Genetic effects can represent i) SNPs that are significantly associated with a phenotype and ii) infinitesimal genetic effects that reflect underlying population structure and relatedness in a cohort. Non-genetic effects are used to simulate environmental, experimental or other unexplained variance in the data. When simulating non-genetic factors, assumptions about their distribution have to be made and this choice depends on the specific biological effects that should be modelled. Common distributions are binomial (e.g. sex), normal or uniform distributions (e.g. weight, height) or categorical variables (e.g disease status).  Correlated non-genetic effects can be used to simulate a phenotype component with a defined level of correlation between traits. For instance, such effects can reflect correlation structure decreasing in phenotypes with ordered or spatial components e.g. in imaging data. Observational noise captures any non-specified effects that arise due to, for instance, experimental measurement error. In addition to the different sources of variation that these components model, they can further differ in their effect distribution across the simulated traits and the proportion of the variance they explain out of the total phenotypic variance. 
%Genetic fixed effects are the effects of interest in genetic association studies i.e. the SNPs that are significantly associated with a phenotype. Genetic effects that reflect underlying population structure and relatedness in a cohort are simulated as random genetic effects. These effects are based on genetic relationship estimates or IBD which can be derived from the samples' genotypes. 
\\
\\
Depending on the scope, there are many ways to simulate these phenotype components. Typically, it is assumed that the overall phenotype structure is well represented by an additive linear combination of individual components \citep{Stephens2013,Marigorta2014,Zhou2014,Loh2014}. In \textit{PhenotypeSimulator}, the phenotypes \( \mat{Y} \inR N P\) of \(N\) samples and \(P\) traits are generated as the sum of i) SNP genetic effects \( \mat{U}  \inR N P\) , ii) infinitesimal genetic effects \( \mat{G} \inR N P\), iii) non-genetic effects \( \mat{C} \inR N P\),  iv) correlated non-genetic effects \( \mat{T} \inR N P\) and  v) observational noise effects \( \mat{\Psi} \inR N P\). For component i-iv, a certain percentage of their variance is shared across all traits (shared) and the remainder is independent (ind) across traits. The option to divide the variance into shared and independent allows for the simulation of phenotypes with additional complexity. For instance, the level of shared and independent fixed genetic effects allows for the simulation of different levels of pleiotropy.
%
\begin{enumerate}
\item \textit{Genetic effects:} For the genetic effects, \(S\) random SNPs for \(N\) samples are drawn from the simulated genotypes. From the \(S\) random SNPs, a proportion \tmat{\theta} is selected to be causal across all traits. \(\matsup{U}{shared}\inR N P\) is simulated as the matrix product of this shared causal SNP matrix \(\matsup{X}{shared}\inR N {\theta \times S}\) and the shared effect size matrix \(\matsup{B}{shared} \inR {\theta  \times S} P\). \tmatsup{B}{shared} in turn is the matrix product of the two normally distributed vectors \(b_s \inR {\theta  \times S} 1\) and \(b_p^T \inR 1 P\). The remaining \((1- \theta ) \times S\) SNPs are simulated to have an independent effect across a limited number of traits \(p^{\text{ind}}\). To realise this structure, \(\matsup{B}{ind} \inR {(1-\theta) \times S} P\) is initialised with normally distributed entries. Subsequently, \(1 - p^{\text{ind}}\) traits are randomly selected and the row entries for \tmatsup{B}{ind} at these traits set to zero. \(\matsup{U}{ind} \inR N P\) is the matrix product of \(\matsup{X}{ind} \inR N {(1 - \theta)  \times S}\) and \tmatsup{B}{ind}.
The genetic effect \tmat{U} is the sum of \tmatsup{U}{shared} and \tmatsup{U}{ind}.

\item \textit{Non-genetic effects:} The non-genetic effects \tmat{C} are based on \(K\) non-genetic covariates \(\mat{F} \inR N K\), with a proportion \(\gamma\) being shared across all traits yielding the shared covariates matrix \(\matsup{F}{shared} \inR N {\gamma \times K} \). The proportion of \(1- \gamma\) non-genetic covariates that are independent make up the independent covariates matrix \(\matsup{F}{ind} \inR N {(1-\gamma) \times K}\). The distributions for each of the \(K\)  non-genetic covariates are independent and can be either normal, uniform, binomial or categorical.  The effect size matrices  \(\matsup{A}{shared} \inR {\gamma \times K} P\)  and \(\matsup{A}{ind}  \inR {(1-\gamma) \times K} P \) were designed as described for the genetic effects. The total non-genetic effect is then \(\mat{C} =  \matsup{F}{shared} \matsup{A}{shared} + \matsup{F}{ind}  \matsup{A}{ind} \).

\item \textit{Infinitesimal genetic effects:} The basis of the infinitesimal genetic effect is the \(N \times N\) genetic relationship matrix \tmat{R}, estimated according to Equation~\cref{eq:relatedness} from the SNP genotypes (of the simulated samples). A suitable model for simulating the infinitesimal genetic effect \(\mat{G} \inR N P\) with known \(N \times N\) sample covariance \tmat{R} and trait covariance \tmat{C} is a multivariate normal distribution where
%
\begin{equation}
 \text{vec}(\mat{G}) \sim \multinormal N P {\text{vec}(\mat{0})} {\mat{C} \otimes \mat{R}}
 \label{eq:G-mn}
\end{equation}
%
The structure of \tmat{C} depends on the desired design of the covariance effect, which can be either shared or independent across traits. This distribution can be realised by simulation a random variable \(\mat{W} \inR K L\) as iid \(\normal 0 1\) and setting 
\begin{equation}
 \text{vec}(\mat{G}) = \mat{BWA}^T
 \label{eq:G-sim}
\end{equation}
where \(\mat{B} \inR N K\) reflects the genetic relationship i.e. sample covariance with \(\mat{R}=\mat{BB}^T\) and \(\mat{A} \inR P L\) the trait covariance with  \(\mat{C}=\mat{AA}^T\), respectively. A detailed derivation for \cref{eq:G-sim} from \cref{eq:G-mn} can be found in \cref{section:simulating-G} in the appendix.

By recasting \cref{eq:G-mn} as \cref{eq:G-sim}, the infinitesimal genetic effect \tmat{G} described by a multivariate-normal distribution is effectively modelled as the product of three matrices, representing the sample covariance (\tmat{B}), a normally distributed variable (\tmat{W}) and the trait covariance (\tmat{A}). Different designs of \tmat{A} will allow for the simulation of shared and independent genetic random effects. For the independent effect, \tmatsup{A}{ind} is a diagonal matrix with normally distributed entries: \((\matsup{A}{ind})^T = \text{diag}(a_1, a_2,  \dotsc , a_P) \sim \normal 0 1\), such that \(\matsup{G}{ind} =  \text{vec}(\mat{BY}(\matsup{A}{ind})^T) \). \tmatsup{A}{shared} of the shared effect is simulated as a matrix of row rank one, with normally distributed entries in row one and zeros elsewhere: \(a_{1,j} \sim \normal 0 1\) and \(a_{i \neq 1,j} = 0\) such that \(\matsup{G}{shared} =  \text{vec}(\mat{BY}(\matsup{A}{shared})^T) \). The total infinitesimal genetic effect \tmat{G} is \(\mat{G} = \matsup{G}{shared} + \matsup{G}{ind}\). 
%
\item \textit{Correlated non-genetic effects:}  Correlated non-genetic effects are simulated as a multivariate normal distribution with a covariance matrix described by the trait-by-trait correlation. The trait-by-trait correlation matrix \tmat{C} is constructed as follows: traits of distance \(d=1\) (adjacent trait columns) will have the highest specified correlation \(r\), traits with \(d=2\) have a correlation of \(r^2\), up to traits with \(d=(P - 1)\) with a correlation of \(r^{(P - 1)})\) , such that the correlation is highest at the first off-diagonal element and decreases exponentially by distance from the diagonal. The final correlated non-genetic effect matrix is simulated as \(\mat{T} \sim \multinormal N P {\mat{0}} {\mat{C}}\).
%
\item \textit{Observational noise:} The observational noise effects \tmat{\Psi} are simulated as the sum of a shared and an independent observational noise effect. The shared effect \tmatsup{\Psi}{shared} is simulated as \(\text{vec}(\matsup{\Psi}{shared}) \sim \normal 0 1\). The independent  effect \tmatsup{\Psi}{ind} is simulated as the matrix product of two normally distributed vectors \(\mat{a} \sim \multinormal N 1 0 1\) and \(\mat{b} \sim \multinormal P 1 0 1\): \(\matsup{\Psi}{ind} = \mat{ab}^T\).
\end{enumerate}
%
\textit{PhenotypeSimulator} enables the specification of the amount of variance that each component should contribute to the total phenotypic variance. Every component is thereby scaled by a factor \(a\) such that its average column variance explains \(x\) percent of the total variance. The scale factor \(a\) is derived as follows: 
Let \(X\) be a random variable with expected value \(E[X] = \mu_{x}\) and variance \(V[X] = E[(X - \mu_{x})^2]\) and let  \(Y = aX\). Then
  %
\begin{equation}
\begin{aligned}
E[Y] &= a\mu_{x} \\
V[Y] &= E[(Y - \mu_{y})^2] \\
V[Y] &= E[(aX - a\mu_{x})^2] \\
		&= a^2 E[(X - \mu_{x})^2]. \\
\end{aligned}
\end{equation}
%
Hence, the scaling of a random variable by \(a\) leads to the scaling of its variance by \(a^2\). To scale the phenotype components such that their average column variance \(\overline{V}_{col} = \frac{V_1 + ... + V_p}{p} \) explains a specified percentage~\(x\) of the total variance, choose the scaling factor \(a\) such that: 
\begin{equation}
\begin{aligned}
x  &= a^2 \times \overline{V}_{col} \\
a  &= \sqrt{x{\overline{V}_{col}}^{-1}}
\end{aligned}
\end{equation}
%
The final simulated phenotype \tmat{Y}  is expressed as the sum of the genetic effects (\tmatsup{U}{scaled}, \tmatsup{G}{scaled}), the non-genetic covariates (\tmatsup{C}{scaled}), correlated non-genetic effects (\tmatsup{T}{scaled}) and observational noise effects (\tmatsup{\Psi}{scaled}):
\begin{equation}
\mat{Y} = \matsup{U}{scaled}   + \matsup{C}{scaled} +  \matsup{G}{scaled} + \matsup{T}{scaled} + \matsup{\Psi}{scaled} .
\end{equation}
%
In \cref{fig:simulation}, I show an example of a simulated phenotype and its different components based on the simulation strategy described above. 
\begin{figure}[h]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter1/Figures/simulatedPhenotypes.pdf}
	\caption[\textbf{Phenotype simulation.}]{\textbf{Phenotype simulation.} Heatmaps of the trait-by-trait correlation (Pearson correlation) of a simulated phenotype ( \tmat{Y}) and its five phenotype components: SNP genetic effects  (\tmatsup{U}{scaled}), infinitesimal genetic effects (\tmatsup{G}{scaled}), non-genetic covariates (\tmatsup{C}{scaled}), correlated non-genetic effects (\tmatsup{T}{scaled}) and observational noise (\tmatsup{\Psi}{scaled}). The non-genetic covariates consist of four independent components, two following a binomial and two following a normal distribution, the SNP genetic effect of ten causal SNPs. The highest correlation for the correlated non-genetic effect was set at \num{0.8}. Apart from the correlated non-genetic component, each component was simulated with \num{80}\% of its variance shared across all traits, while the rest remained independent. The total genetic variance accounted to \num{60}\% leaving \num{40}\% of variance explained by the non-genetic terms.}
	\label{fig:simulation}
\end{figure}

%The simulation of complex phenotypes is a common place task in methodological development. Simulations by necessity have to have an underlying model and are not free of assumptions. The simulation strategy described here accounts for complex population structure, random fixed genetic and noise effects and produces challenging phenotypic distributions. In order to broadly distribute this simulation framework, I have developed \textit{PhenotypeSimulator}, an easily accessible tool for phenotype simulation that allows for a flexible and customisable simulation set-up. PhenotypeSimulator can be installed from the Comprehensive R Archive Network \citep{Meyer2017b} and is currently under review for publication \citep{Meyer2017a}.
The phenotype consists of five traits for each of the \num{1000} samples from a cohort of related individuals with no population structure. There are a total of ten causal SNPs and four covariates associated with the phenotype. In addition, it is composed of background genetic and noise effects as well as a correlated non-genetic effect (correlation: \num{0.8}). The total genetic variance accounts for \num{60}\% of the variance leaving \num{40}\% of variance explained by the noise terms.

Phenotypes simulated with \textit{PhenotypeSimulator} built the basis for the method development in the following chapter.