\chapter{PhenotypeSimulator}
\label{chapter:simulation} 
Often, a first step in new method development is to reverse the task of genotype to phenotype mapping. The latter is commonly realised by fitting a linear model to the phenotype measurements and treating the genotype and possible covariates as explanatory variables. In order to evaluate new methods, one needs to have a set of well-characterised genotypes and phenotypes to know the ground truth based on which comparisons of the model performance can be made. Based on the underlying linear association methods, the phenotypes are often simulated as a linear composition of different effect components. 

For the methods development and selection process in this thesis (\cref{section:limmbo} and \cref{section:dimreduction}), I needed well-characterised phenotypes for evaluation before applying the methods to real datasets and biological questions. In the following section, I will first describe the simulation of genotypes with different levels of population structure and relatedness, followed by a description of the phenotype simulation. The simulation strategies described in this section apply to all simulated datasets within this thesis. 

\section{Genotype simulation}
\label{section:genotype-simulation}
Genotypes can either be generated by sampling from real data sets or by simulating SNPs anew. In the latter case, assuming bi-allelic SNPs, each SNP is simulated from a binomial distribution with two trials and probability equal to the given allele frequencies  This simple approach, however, does not account for any LD structure in the genome. In contrast, sampling from a diverse set of real genotypes does not only facilitate retaining realistic LD structure in the genotypes, it also allows for the simple simulation of more defined population structure and relatedness within a cohort. As I wanted to evaluate the methods described in this thesis across different levels of relatedness and population structure, 
I chose to simulate the genotypes based on real genotype data from four European ancestry populations of the 1000 Genomes (1KG) Project (populations: CEU, FIN, GBR, TSI) \citep{Abecasis2012}, similar to strategies described in \citep{Loh2014,Casale2015}. I simulated three genotype sets, each with 1,000 samples, that differed i) in the number of ancestors \(N\) from which the genotypes were chosen and ii) the subpopulations the ancestors were chosen from:
\begin{enumerate}
\item unrelatedPopStructure: unrelated individuals with prior assigment of ancestral population  (\(N=10\))
\item unrelatedNoPopStructure: unrelated individuals with mixed ancestral population  (\(N=10\))
\item relatedNoPopStructure: related individuals with mixed ancestral population (\(N=2\))
\end{enumerate}

The number of ancestors sets the the level of relatedness within a cohort, where low numbers of \(N\) introduce relatedness among individuals, while high numbers of \(N\) lead to low levels of structure and relatedness. The choice of ancestrial population determines the level of subpopulation formation in the simulated genotypes: allowing for random selection of ancestors independent of the four subpopulations in the 1KG datasets yields low levels of population structure, whereas choosing ancestors based on their subpopulation gives rise to subpopulations in the simulated dataset as well. 
For the simulation, each newly synthesised individual is assigned to \(N\) ancestors from the original 1KG Project and their genome split into blocks of 1,000 SNPs. For each SNP block, an ancestor is chosen at random either from the whole dataset (NoPopstructure) or a subpopolution (Popstructure)  and its genotype is copied to the individuals genome. 

The level of structure and relatedness introduced by this simulation strategy can be visualised by examining the GRM and the principal components of the genotypes. The kinship as estimated via \cref{eq:kinship} is a measure of relatedness between the individuals, while principal components reflect the genotypic variance in the data. The hierarchical clustering of the genetic relationship estimates and scatter plots of the first two principal components for each genotype set are shown in \cref{fig:kinship-matrices}. Samples cluster tightly based on their ancestrial subpopulations (\cref{fig:kinship-matrices}\subfig{A}), while there is no clustering and an even spread in the PC plot for the cohort of unrelated individuals with ancestors sampled across all subpopulations (\cref{fig:kinship-matrices}\subfig{B}). The cohort of related individuals shows less spread in the second principal component and higher individual genetic relationship  estimates (\cref{fig:kinship-matrices}\subfig{C}).


\begin{figure}[htbp]
	\centering
	\includegraphics[page=2, trim = 0mm 10mm 15mm 0mm, clip, scale=0.25]{Chapter1/Figures/simulatedCovarianceMatrices_kinship.png}
	\caption[\textbf{Genetic relationship matrices and principal components of three simulated European ancestry cohorts.}]{\textbf{Genetic relationship matrices and principle components of three simulated European ancestry cohorts.} The genotypes were simulated based in genotype data from four European ancestry populations (ancestry colour key in panel A). Depending on the choice and number of ancestors for the sampling of chromosomes to simulate an individual's genotype, cohorts with differing levels of population and relatedness structure will arise. The left column depicts the hierarchical clustering of the sample-to-sample  genetic relationship coefficients (complete linkage clustering of euclidean distance between coefficients), the right column the first and second principal component (PC) of the sample genotypes for the three different cohorts: A. unrelated individuals, with population structure: \(N=10\), prior assigment to ancestral population; B. unrelated individuals, no population structure: \(N=10\), no prior assigment to ancestral population; C. related individuals, no population structure: \(N=2\), no prior assigment to ancestral population.}
 	\label{fig:kinship-matrices}
\end{figure}


\section{Phenotype simulation}
\label{section:phenotype-simulation}
The simulation of phenotypes and their components can be described on two levels, either based on the biological meaning they reflect or the statistical type of the effect. In statistics, one distinguishes the fixed effects which are constant across individuals, from random effect which can vary (discussed in detail in \citep{Gelman2005}). On the biological level, we can classify the phenotypic components into genetic and non-genetic (noise) components.
Commonly simulated phenotype components are fixed and random genetic effects and fixed, correlated and random noise effects (e.g.\citep{Stephens2013,Marigorta2014,Zhou2014,Loh2014}). 

Genetic fixed effects are the effects of interest in genetic association studies i.e. the SNPs that are significantly associated with a phenotype. Genetic effects that are not associated on a per-SNP basis but reflect underlying population structure and relatedness in a cohort are simulated as random genetic effects. These effects are based on genetic relationship estimates or IBD which can be derived from the samples' genotypes. Non-genetic effects are used to simulate environmental, experimental or noise effects. Fixed noise effects are use to simulate confounding variables or covariates in an analysis, such as sex, age, weight or disease status. When simulating such confounding structures, assumptions about their distribution have to be made and this choice depends on the specific biological effects that should be modeled. Common distribution are binomial (e.g. sex), normal or uniform distribution (e.g. weight, height) or categorical (e.g disease status). Random noise effects simulate any non-specified noise effects that could arise due to, for instance, experimental measurement error.  Correlated noise effects are a type of random effect that can be used to simulate a phenotype component with a defined level of correlation between traits. For instance, such effects can reflect correlation structure decreasing in phenotypes with ordered or spatial components e.g. in imaging data. 

In addition to the different sources of variation these components model, they can further differ in their effect distribution across the simulated traits and the proportion of the variance they explain out of the total phenotypic variance. The simulation strategy of these components, the effect distribution and the scaling of the components to a specifc proportion of variance are described below. 
\\
\\
The phenotypes \( \mat{Y} \inR N P\) of \(N\) samples and \(P\) traits are generated as the sum of i) fixed genetic effects \( \mat{U}  \inR N P\) , ii) random genetic effects \( \mat{G} \inR N P\), iii) fixed noise effects \( \mat{C} \inR N P\), iv) random noise effects \( \mat{\Psi} \inR N P\) and v) correlated noise effects \( \mat{T} \inR N P\). For component i-iv, a certain percentage of their variance is shared across all traits (shared) and the remainder is independent (ind) across traits.

\begin{enumerate}
\item \textit{Fixed genetic effects:} For the fixed genetic effects, \(S\) random SNPs for \(N\) samples are drawn from the simulated genotypes. From the \(S\) random SNPs, a proportion \tbm{\theta} is selected to be causal across all traits.\(\matsup{U}{shared}\inR N P\) is simulated as the matrix product of this shared causal SNP matrix \(\matsup{X}{shared}\inR N {\theta \times S}\) and the shared effect size matrix \(\matsup{B}{shared} \inR {\theta  \times S} P\) . \tmatsup{B}{shared} in turn is the matrix product of the two normally distributed vectors \(b_s \inR {\theta  \times S} 1\) and \(b_p^T \inR 1 P\). The remaining \((1- \theta ) \times S\) SNPs are simulated to have an independent effect across a limited number of traits \(p^{\text{ind}}\). To realise this structure, \(\matsup{B}{ind} \inR {(1-\theta) \times S} P\) is initialised with normally distributed entries. Subsequently, \(1 - p^{\text{ind}}\) traits are randomly selected and the row entries for \tmatsup{B}{ind} at these traits set to zero. \(\matsup{U}{ind} \inR N P\) is the matrix product of \(\matsup{X}{ind} \inR N {(1 - \theta)  \times S}\) and \tmatsup{B}{ind}.
The fixed genetic effect \tmat{U} is the sum of \tmatsup{U}{shared} and \tmatsup{U}{ind}.

\item \textit{Fixed noise effects:} The fixed noise effects \tmat{C} are based on \(K\)  confounders \(\mat{F} \inR N K\), with a proportion \(\gamma\) being shared across all traits yielding the shared confounder matrix \(\matsup{F}{shared} \inR N {\gamma \times K} \). The proportion of \(1- \gamma\) confounder that are independent make up the independent confounder matrix \(\matsup{F}{ind} \inR N {(1-\gamma) \times K}\). The distributions for each of the \(K\)  confounders are independent and can be either normal, uniform, binomial or categorical.  The effect size matrices  \(\matsup{A}{shared} \inR {\gamma \times K} P\)  and \(\matsup{A}{ind}  \inR {(1-\gamma) \times K} P \) were designed as described for the fixed genetic effects. The total fixed noise effect is then \(\mat{C} =  \matsup{F}{shared} \matsup{A}{shared} + \matsup{F}{ind}  \matsup{A}{ind} \).

\item \textit{Random genetic effects:} The random genetic effects \(\mat{G} \inR N P\) are modeled as a matrix-normally distributed random variable, defined by its mean \(\mat{M} \inR N P\), its column covariance \(\mat{C} \inR P P\) and its row covariance \(\mat{D} \inR N N\): 

\begin{equation}
\mat{G} \sim \matrixnormal N P M D C.
\end{equation}

The \(N \times N\) genetic relationship matrix \tmat{R}, estimated according to Equation~\ref{eq:kinship} from the SNP genotypes (of the simulated samples) represents the row covariance \tmat{D}.  The structure of the trait-to-trait covariance \tmat{C} depends on the design of the covariance effect, which can be either shared or independent across traits. To construct \tmat{G} from shared and independent random genetic effects, assume a matrix-normally distributed random variable \tmat{Z} with \(\mat{M}=0\)  and \(\mat{D} = \mat{R}\):

\begin{equation}
\mat{Z} \sim  \matrixnormal N P 0 R C
\end{equation}

\tmat{Z} can be expressed in terms of a multivariate normal distribution 

\begin{equation}
\text{vec}(\mat{Z}) \sim \multinormal N P {\mat{0}} {\mat{C} \otimes \mat{R}}.
\end{equation}

With the cholesky decompositon of \tmat{K} and \tmat{C} into  \(\mat{E}=\mat{BB}^T\) and \(\mat{C}=\mat{AA}^T\) 

\begin{equation}
\text{vec}(\mat{Z}) \sim \multinormal N P {\mat{0}} {\mat{AA}^T  \otimes \mat{BB}^T},  
\end{equation}

which can be rearranged as 

\begin{equation}
\begin{aligned}
\text{vec}(\mat{Z}) \sim \multinormal N P {\mat{0}} {(\mat{A} \otimes \mat{B}) \mat{I} (\mat{A}^T \otimes \mat{B}^T)} \\
\text{vec}(\mat{Z})  \sim \multinormal N P {\mat{0}} {(\mat{A }\otimes \mat{B}) \mat{I} (\mat{A} \otimes\mat{B})^T)}, 
\end{aligned}
\end{equation}
with \tmat{I} as the identity matrix.
 
Using the property of a normally distributed random variable \tmat{Y} with mean \tmat{\mu} and covariance matrix \tmat{\Sigma}

\begin{equation}
\begin{aligned} 
w \mat{Y} \sim \normal {w\mat{\mu}}  {w\mat{\Sigma} w^T},
\end{aligned}
\end{equation}

we can let  \(\text{vec} (\mat{Z}) =  (\mat{A} \otimes \mat{B}) \text{vec} (\mat{Y})\)  and \(\mat{Y} \sim \multinormal N P {\mat{0}} {\mat{I}}\) such that

\begin{equation}
\begin{aligned}
(\mat{A} \otimes\mat{B}) \text{vec} (\mat{Y})  \sim \multinormal N P {\mat{0}} {(\mat{A} \otimes \mat{B}) \mat{I} (\mat{A} \otimes \mat{B})^T}
\end{aligned}
\end{equation}

Using \citep{Horn1991}: Lemma 4.3.1, we get 
\begin{equation}
(\mat{A} \otimes \mat{B}) \text{vec}(\mat{Y}) = \text{vec}(\mat{BYA}^T) 
\end{equation}

For the independent effect, \tmatsup{A}{ind} is a diagonal matrix with normally distributed entries: \((\matsup{A}{ind})^T = \text{diag}(a_1, a_2,  \dotsc , a_P) \sim \normal 0 1\), such that \(\matsup{G}{ind} =  \text{vec}(\mat{BY}(\matsup{A}{ind})^T) \). \tmatsup{A}{shared} of the shared effect is a matrix of row rank one, with normally distributed entries in row 1 and zeros elsewhere: \(a_{1,j} \sim \normal 0 1\) and \(a_{i \neq 1,j} = 0\) such that \(\matsup{G}{shared} =  \text{vec}(\mat{BY}(\matsup{A}{shared})^T) \). The total random genetic effect \tmat{G} is \(\mat{G} = \matsup{G}{shared} + \matsup{G}{ind}\). 

\item \textit{Random noise effects:} The random noise effects \tmat{\Psi} are simulated as the sum of a shared and an independent random noise effect. The shared random effect \tmatsup{\Psi}{shared} is simulated as \(\text{vec}(\matsup{\Psi}{shared}) \sim \normal 0 1\). The independent random effect \tmatsup{\Psi}{ind} is simulated as the matrix product of two normally distributed vectors \(\mat{a} \sim \multinormal N 1 0 1\) and \(\mat{b} \sim \multinormal P 1 0 1\): \(\matsup{\Psi}{ind} = \mat{ab}^T\).


\item \textit{Correlated noise effects:}  Correlated noise effects are simulated as a multivariate normal distribution with a covariance matrix described by the trait-trait correlation. The trait-trait correlation matrix \tmat{C} is constructed as follows: traits of distance \(d=1\) (adjacent trait columns) will have the highest specified correlation \(r\), traits with \(d=2\) have a correlation of \(r^2\), up to traits with \(d=(P - 1)\) with a correlation of \(r^{(P - 1)})\) , such that the correlation is highest at the first off-diagonal element and decreases exponentially by distance from the diagonal. The final correlated noise effect matrix is simulated as \(\mat{T} \sim \multinormal N P {\mat{0}} {\mat{C}}\).
\end{enumerate}

Before combining the different components into the final phenotype, each component is rescaled by a factor \(a\) such that their average column variance explains \(x\) percent of the total variance. The scale factor \(a\) is derived as follows: 
Let \(X\) be a random variable with expected value \(E[X] = \mu_{x}\) and variance \(V[X] = E[(X - \mu_{x})^2]\) and let  \(Y = aX\). Then
  
\begin{equation}
\begin{aligned}
E[Y] &= a\mu_{x} \\
V[Y] &= E[(Y - \mu_{y})^2] \\
V[Y] &= E[(aX - a\mu_{x})^2] \\
		&= a^2 E[(X - \mu_{x})^2]. \\
\end{aligned}
\end{equation}

Hence, the scaling of a random variable by \(a\) leads to the scaling of its variance by \(a^2\). To scale the phenotype components such that their average column variance \(\overline{V_{col}} = \frac{V_1 + ... + V_p}{p} \) explains a specified  percentage \(x\) of the total variance, choose the scaling factor \(a\) such that: 
\begin{equation}
\begin{aligned}
x  &= a^2 \times \overline{V_{col}} \\
a  &= \sqrt{\frac{x}{\overline{V_{col}}}}
\end{aligned}
\end{equation}

The final simulated phenotype is expressed as
\begin{equation}
\mat{Y} = \matsup{U}{scaled}   + \matsup{C}{scaled} +  \matsup{G}{scaled} +  \matsup{\Psi}{scaled} + \matsup{T}{scaled}. 
\end{equation}

In \cref{fig:simulation}, I show an example of a simulated phenotype and its different components based on the simulation strategy described above. The phenotype consists of five traits for each of the \num{1000} samples from a cohort of related individuals with no population structure. There are a total of ten causal SNPs and four covariates associated with the phenotype. In addition, it is composed of background genetic and noise effects as well as a correlated noise effect (correlation: \num{0.8}). The total genetic variance accounts for \num{60}\% of the variance leaving \num{40}\% of variance explained by the noise terms.

\begin{figure}[hbtp]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=\textwidth]{Chapter1/Figures/simulatedPhenotypes.png}
	\caption[\textbf{Phenotype simulation.}]{\textbf{Phenotype simulation.} Heatmaps of the trait-to-trait correlation (Pearson correlation) of a simulated phenotype and its five phenotype components: fixed (fixedGenetic) and random (randomGenetic) genetic effects and fixed (fixedNoise), random (randomNoise) and correlated (correlatedNoise) noise effects. The fixed noise effects consist of four independent components, two following a binomial and two following a normal distribution, the fixed genetic effect of ten causal SNPs. The highest correlation for the correlated noise effect was set at \num{0.8}. Apart from the correlated noise component, each component was simulated with \num{80}\% of its variance shared across all traits, while the rest remained independent. The total genetic variance accounted to \num{60}\% leaving \num{40}\% of variance explained by the noise terms.}
	\label{fig:simulation}
\end{figure}


Developing new methods in quantitative genetics often requires simulated datasets with a well-characterised phenotype structure. Thereby, the number of phenotype components and their contribution to the final phenotypic variants depend on the task at hand. In order to provide a tool for phenotype simulation that is easily accessible and allows flexible simulation set-ups, I turned this simulation framework into the R package \textit{PhenotypeSimulator}, which can be installed from the Comprehensive R Archive Network \citep{Meyer2017} \red{describe in more detail?}.

