\chapter{PhenotypeSimulator}
\label{chapter:simulation} 
For method development in quantitative genetics, one often needs a set of well-characterised genotypes and phenotypes to know the ground truth based on which comparisons of the model performance can be made. In the context of this thesis, genotype and phenotype simulations were crucial for the development of a new method for multi-trait mapping of high-dimensional datasets (\cref{chapter:limmbo}) and the evaluation of different dimensionality reduction techniques (\cref{chapter:DimReduction}).

The complexity of the simulated phenotype components will depend on the specifics of the model that is being developed. With the detailed whole-genome genotype data available through standard techniques such as genotyping arrays and subsequent imputation and the measurement of multiple traits per sample, the complexity of the hypotheses for testing the underlying genetics of the observed phenotypes have increased. Models range from simple linear models with a few fixed effects on a single trait to complex linear mixed models with fixed and random effect components on multiple traits \citep{Stephens2013,Marigorta2014,Zhou2014,Loh2014}. With the increase in analysis complexity, sophisticated approaches for modelling realistic genotype and phenotype structures are needed. These simulated genotypes and phenotypes reflect our perceived understanding of the true phenotype structure and do not guarantee the biologically correctness of real phenotypes. However, they are invaluable in model design, as any model showing flawed statistics on the possibly simplified biological model will suffer from at least the same flaws on the true biological data.

In this chapter, I will first describe simulation strategies for genotypes with different levels of population structure and relatedness. Following that, I introduce the phenotype simulation strategy used for all simulated datasets within this thesis. In order to broadly distribute this simulation framework, I have developed \textit{PhenotypeSimulator}, an R package for phenotype simulation that allows for a flexible and customisable simulation set-up. \textit{PhenotypeSimulator} can be installed from the Comprehensive R Archive Network \citep{Meyer2017} and its code is available on github: \url{https://github.com/HannahVMeyer/PhenotypeSimulator}. \textit{PhenotypeSimulator} is published as: Meyer, H. \& Birney E. (2018) PhenotypeSimulator: A comprehensive framework for simulating multi-trait, multi-locus genotype to phenotype relationships, \textit{Bioinformatics}, bty197. 

\section{Genotype simulation}
\label{section:genotype-simulation}
There are a number of different strategies to generate genotype data for genetic association studies. In the most simple case and assuming bi-allelic \glspl{snp}, each \gls{snp} is simulated from a binomial distribution with two trials and probability equal to the given allele frequencies  (e.g in \citep{Lippert2013}). This simple approach, however, does not simulate any dependency between the genotypes as is observed with \gls{ld} structure in the genome. In order to mimic genomic \gls{ld} structure and allele frequency distributions in the simulated dataset, three general approaches exist: i) backward-time or coalescent simulation, ii) forward time and iii) resampling approaches. The coalescent \citep{Hudson2002,Ewing2010,Kelleher2016} and forward-time approaches \citep{Peng2007,Hoggart2007,Carvajal-Rodriguez2008} use population genetic models to simulate genotypes and are particularly useful for studying evolution and demography. However, they often suffer from computational demands for diploid genome-wide \gls{snp} data \citep{Liu2008,Yuan2012}. Resampling approaches \citep{Wright2007,Su2011,Loh2014,Casale2015} offer a practical solution that can be used to efficiently generate genetic data with different relatedness and population structures, which is particularly useful in genetic association studies. They combine existing genotype data into the genotypes of the simulated samples, thereby retaining allele frequency and \gls{ld} patterns. 

I choose to follow the resampling strategies described in \citep{Loh2014,Casale2015} where each diploid individual is simulated as the mosaic of real genotypes from different populations.  Depending on the simulation set-up, cohorts with differing levels of population structure and relatedness can be simulated. Cohorts with different degrees of genetic structure will be valuable for evaluating the performance of genetic association models with respect to their adjustment for genetic relatedness and population structure.  As far as I am aware, these structures cannot be realised with the publicly available tools described in \citep{Wright2007,Su2011}.

I used the genotype data from \num{365} individuals of four European ancestry populations from the 1000 Genomes Project \citep{1000Genomes2015}, \gls{ceu} and \gls{fin} and  \gls{gbr} and \gls{tsi}, as the sampling dataset.  The resampling strategy works as follows:
\begin{enumerate}
\item each individual is randomly assigned a predefined number of unique original genotypes which will serve as its ancestors;
\item the ancestors' genome-wide genotypes are split into blocks of \num{1000} \glspl{snp};
\item  for each \gls{snp} block, one of the ancestor is chosen at random and its genotype is copied to the individual's genome. 
\end{enumerate}

The number and the sub-population of ancestors that are chosen for simulating the genomes of a new cohort are critical for controlling the level of structure within the cohort. 

The number of ancestors sets the level of relatedness within the cohort. Low numbers of \(N\) introduce relatedness among individuals, while high numbers of \(N\) lead to low levels of structure and relatedness. For instance, with \(N=2\), each individual in the newly synthesised cohort is composed of genotypes from only two out of the \num{365} individuals. Consider individual \(g_1\), whose genotypes are drawn from ancestors \(a_1\) and \(a_2\). For Individual \(g_2\), with a chance of \(p=1- \frac{\binom{363}{2}}{\binom{365}{2}} \approx 0.01\) it shares at least one ancestor with \(g_1\). For exactly one shared ancestor, each \gls{snp} block would have a \num{25}\% probability of being the same between  \(g_1\) and  \(g_2\) . With \(N=10\), the probability for at least one common ancestor increases (\(p=1 - \frac{\binom{355}{10}}{\binom{365}{10}} \approx 0.25\)). However, for exactly one shared ancestor, the sharing of \gls{snp} blocks decreases to \num{1}\%. 

The choice of sub-population determines the level of population structure in the simulated genotypes: allowing for random selection of ancestors independent from the four subpopulations in the 1000 Genomes datasets yields low levels of population structure, as this leads to a random sampling of the individuals' genotypes across ancestors ethnicities. Including an \textit{a priori} selection of individuals from one of the four sub-population and subsequently restricting ancestor selection to these individuals will restrict an individuals genotypes to a single sub-population.  As all individuals in the cohort are now comprised of distinct genotype subsets, this will give rise to population structure in the simulated cohort. 

I simulated three genotype sets, each with \num{1000} samples, that differed i) in the number of ancestors \(N\) from which the genotypes were chosen and ii) the sub-popula- tions the ancestors were chosen from:
\begin{enumerate}[label=\Alph*.]
\item unrelatedPopStructure: unrelated individuals with prior assignment of ancestral population  (\(N=10\), i.e. only \gls{ceu} or only \gls{fin} or only \gls{gbr} or only \gls{tsi})
\item unrelatedNoPopStructure: unrelated individuals with mixed ancestral population  (\(N=10\), i.e. \gls{ceu} and \gls{fin} and  \gls{gbr} and \gls{tsi})
\item relatedNoPopStructure: related individuals with mixed ancestral population (\(N=2\), i.e. \gls{ceu} and \gls{fin} and  \gls{gbr} and \gls{tsi}))
\end{enumerate}


The level of structure and relatedness introduced by this simulation strategy can be visualised by examining the genetic relationship matrix and the \glspl{pc} of the genotypes. The genetic relationship matrix is estimated as a \gls{rrm} via \cref{eq:relatedness} and serves as a measure for relatedness between the individuals, while \glspl{pc} reflect the genotypic variance in the data (\cref{subsubsection:relatedness-model-variables}). The hierarchical clustering of the genetic relationship estimates and scatter plots of the first two \glspl{pc} for each genotype set are shown in \cref{fig:kinship-matrices}. Samples cluster tightly based on their ancestral subpopulations (\cref{fig:kinship-matrices}\subfig{A}), while there is no clustering and an even spread in the \gls{pc} plot for the cohort of unrelated individuals with ancestors sampled across all subpopulations (\cref{fig:kinship-matrices}\subfig{B}). The cohort of related individuals shows less spread in the second principal component and higher individual genetic relationship estimates (\cref{fig:kinship-matrices}\subfig{C}).


\begin{figure}[htbp]
	\centering
	\includegraphics[page=2, trim = 0mm 10mm 15mm 0mm, clip, scale=0.25]{Chapter1/Figures/simulatedCovarianceMatrices_kinship.png}
	\caption[\textbf{Genetic relationship matrices and principal components of three simulated European ancestry cohorts.}]{\textbf{Genetic relationship matrices and principal components of three simulated European ancestry cohorts.} The genotypes were simulated based on genotype data from four European ancestry populations (ancestry colour key in panel A). Depending on the choice and number of ancestors for the sampling of chromosomes to simulate an individual's genotype, cohorts with differing levels of population and relatedness structure will arise. The left column depicts the hierarchical clustering of the sample-by-sample  genetic relationship coefficients (complete linkage clustering of Euclidean distance between coefficients), the right column the first and second \gls{pc} of the sample genotypes for the three different cohorts: A. unrelated individuals, with population structure: \(N=10\), prior assignment to ancestral population. B. unrelated individuals, no population structure: \(N=10\), no prior assignment to ancestral population. C. related individuals, no population structure: \(N=2\), no prior assignment to ancestral population.}
 	\label{fig:kinship-matrices}
\end{figure}

\newpage
\section{Phenotype simulation}
\label{section:phenotype-simulation}
In this section, I introduce \textit{PhenotypeSimulator}, an R/CRAN package for the flexible simulation of phenotypes with different genetic and non-genetic variance components. \textit{PhenotypeSimulator} is a framework focusing on the simulation of phenotypes, with a particular emphasis on complexity of both multiple phenotypes and multiple genetic loci and genetic background, which is not provided by other multi-phenotype simulation software (\citep{OReilly2012}, \citep{Porter2017}).  I have written \textit{PhenotypeSimulator} to be easily integrated with external genotype simulation software (such as coalescent and forward time simulation and re-sampling approaches) and it can generate output suitable as input for a number of standard genetic association tools (such as PLINK \citep{Chang2015}, GEMMA \citep{Zhou2014} or SNPTEST \citep{Marchini2007}). In the following, I will describe the simulation strategy of the different phenotype components, and will demonstrate the usage and application of \textit{PhenotypeSimulator} by simulating phenotypes to evaluate the power of different linear mixed model designs in a genetic association study.
%
\begin{figure*}[hbtp]
	\centering
	\includegraphics[width=\textwidth]{Chapter1/Figures/SimulationScheme.pdf}
	\caption[\textbf{Phenotype simulation scheme.}]{\textbf{Phenotype simulation scheme.} \textit{PhenotypeSimulator} can take genotypes from a number of different input formats and uses these as the basis for the simulation of the genetic effects. In addition to the genetic effects, non-genetic covariates, observational noise and non-genetic correlation structure can be simulated. The effect structure of the upper four components can be divided into a shared effect across traits or an independent effect for a number of traits, allowing for complex phenotype structures such as the simulation of pleiotropy. Before combining the phenotype components, they are scaled to a user-defined proportion of the total phenotypic variance. Finally, the simulated phenotype and its components can be saved into a number of different genetic output formats. Arrows, lines and rectangle mark the dimensions of each component.} 
	 	\label{fig:simulation-scheme}
\end{figure*}


%\textit{PhenotypeSimulator} can simulate simple bi-allelic SNPs, where each SNP is simulated from a binomial distribution with two trials and probability equal to the given allele frequencies (as for instance used in \citep{Lippert2013}). This simple approach, however, does not simulate any dependency between the genotypes as is observed with LD structure in the genome. To allow for more complex genotype structures, \textit{PhenotypeSimulator} can import genotypes  generated from different genotype simulation software, covering genotypes simulated from coalescent models (GENOME \citep{Liang2007}), a simple re-sampling based approach (HAPGEN2 \citep{Su2011}) and different forward-time approaches (delimited-formats as given by simuPOP \citep{Peng2005}, ForSim \citep{Lambert2008}, GenomePop2 \citep{Carvajal-Rodriguez2008}). In addition, standard genotype formats such as PLINK \citep{Chang2015} or BIMBAM \citep{Guan2008} are supported.

Phenotypes are typically generated as the sum of genetic effects, effects from non-genetic factors and observational noise. Genetic effects can represent i) genetic variants that are associated with a phenotype and ii) infinitesimal genetic effects that reflect underlying population structure and relatedness in a cohort. Non-genetic effects are used to simulate environmental, experimental or other unexplained variance in the data. 
Although in many genetic association studies the sources of non-genetic correlation are often combined, I have found it valuable to separate these components to explore the impact of different correlation structures from these sour- ces (see \cref{chapter:DimReduction}).
When simulating non-genetic factors, assumptions about their distribution have to be made and this choice depends on the specific biological effects that should be modelled. Common distributions are binomial (e.g. sex), normal or uniform distributions (e.g. weight, height) or categorical variables (e.g disease status).  Correlated non-genetic effects can be used to simulate a phenotype component with a defined level of correlation between traits. For instance, such effects can reflect correlation structure decreasing in phenotypes with ordered or spatial components e.g. in imaging data. Observational noise captures any non-specified effects that arise due to, for instance, experimental measurement error. However, \textit{PhenotypeSimulator} can also be used with a combined non-genetic covariance model, similar to more standard linear mixed models \citep{OReilly2012,Zhou2014,Porter2017}

The proportion of variance assigned to each component will differ depending on the biological understanding of the simulated phenotype. \textit{PhenotypeSimulator} allows for the specification of these variance proportions and, in addition, provides the option to divide the explained variance into two components, one that is shared across phenotypes and a second component that acts independently on certain phenotypes. For instance, the level of shared and independent effects for a genetic variant allows for the simulation of different levels of pleiotropy.

There are many ways to simulate these phenotype components depending on the scope and the model to be tested. Typically, it is assumed that the overall phenotype structure is well represented by an additive linear combination of individual components \citep{Stephens2013,Marigorta2014,Zhou2014,Loh2014}. For \textit{PhenotypeSimulator}, I assume this phenotype structure and sum the individual phenotype components to generate the final phenotypes.

\subsection{Phenotype components}
 In \textit{PhenotypeSimulator}, the phenotypes \( \mat{Y} \inR N P\) of \(N\) samples and \(P\) traits are generated as the sum of i) genetic variant effects \( \mat{U}  \inR N P\) , ii) infinitesimal genetic effects \( \mat{G} \inR N P\), iii) non-genetic effects \( \mat{C} \inR N P\),  iv) correlated non-genetic effects \( \mat{T} \inR N P\) and  v) observational noise effects \( \mat{\Psi} \inR N P\) (\cref{fig:simulation-scheme}). For component i-iv, a certain percentage of their variance is shared across all traits (shared) and the remainder is independent (ind) across traits. The option to divide the variance into shared and independent allows for the simulation of phenotypes with additional complexity. For instance, the level of shared and independent fixed genetic effects allows for the simulation of different levels of pleiotropy.

\begin{enumerate}
\item \textit{Genetic variant effects:} For the SNP genetic effects, \(S\) random SNPs for \(N\) samples are drawn from the (simulated) genotypes. From the \(S\) random SNPs, a proportion \tmat{\theta} is selected to be causal across all traits. The shared genetic variant effect is simulated as the matrix product of this shared causal SNP matrix \(\matsup{X}{shared}\inR N {\theta \times S}\) and the shared effect size matrix \(\matsup{B}{shared} \inR {\theta  \times S} P\). The columns of the shared effect size matrix are simulated to be perfectly correlated, i.e. the effect of a SNP genetic effect is proportionally the same for all affected traits. The effect sizes for \(\matsup{B}{shared}\) can either be simulated to have normal or uniform properties. The is implemented as follows in \textit{PhenotypeSimulator}:
\tmatsup{B}{shared} is the matrix product of the two vectors \(b_s \inR {\theta  \times S} 1\) and \(b_p^T \inR 1 P\). To simulate effect sizes with approximately normal properties \citep[Eq 31-33]{Oliveira2012}, \(b_s\) and \(b_p\) are drawn from two normal distributions, where \(\mu_{b_p}=0\) and \(\sigma_{b_p}=1\) and \(\mu_{b_s}\) and \(\sigma_{b_s}\) specified by the user. For the simulation of uniformly distributed effect sizes,  \(b_s \) and \(b_p^T\) are drawn from two exponential distributions whose negative normalised log product yields an approximate uniform distribution \citep{Song2005} across the user defined range. 
The remaining \((1- \theta ) \times S\) SNPs are simulated to have an independent effect across a specified number of traits \(P^{\text{ind}}\). To realise this structure, \(\matsup{B}{ind} \inR {(1-\theta) \times S} P\) is initialised with either normally or uniformly distributed entries, with \(\mu_{B}\) and \(\sigma_{B}\) as specified by the user (same as for shared effect). Subsequently, \(P - P^{\text{ind}}\) traits are randomly selected and the row entries for \tmatsup{B}{ind} at these traits set to zero. The independent genetic variant effect is the matrix product of \(\matsup{X}{ind} \inR N {(1 - \theta)  \times S}\) and \tmatsup{B}{ind}.


\item \textit{Non-genetic covariate effects:} The non-genetic covariate effects are based on \(K\) non-genetic covariates \(\mat{W} \inR N K\), with a proportion \(\gamma\) being shared across all traits yielding the shared covariates matrix \(\matsup{W}{shared} \inR N {\gamma \times K} \). The proportion of \(1- \gamma\) non-genetic covariates that are independent make up the independent covariates matrix \(\matsup{W}{ind} \inR N {(1-\gamma) \times K}\). The distributions for each of the \(K\)  non-genetic covariates are independent and can be either normal, uniform, binomial or categorical. The distribution and respective parameters are chosen by the user. The effect size matrices  \(\matsup{A}{shared} \inR {\gamma \times K} P\)  and \(\matsup{A}{ind}  \inR {(1-\gamma) \times K} P \) were designed as described for the genetic effects. The final non-genetic covariate effects are the matrix product of the covariate matrices and their effect size matrices: \(\matsup{W}{ind}\matsup{A}{ind}\) and \(\matsup{W}{shared}\matsup{A}{shared}\).

\item \textit{Infinitesimal genetic effects:} The basis of the infinitesimal genetic effect \tmat{U} is the \(N \times N\) genetic relationship matrix \tmat{K}, either estimated from the genotypes of the simulated samples as \(\frac{1}{m}\mat{XX}^T\), where \(m\) is the mean value of the diagonal elements of  \(\mat{XX}^T\) or provided by the user. A suitable model for simulating the infinitesimal genetic effect \(\mat{U} \inR N P\) with the known  \(N \times N\) sample covariance \tmat{K} and trait covariance \tmat{C} is a multivariate normal distribution (as for instance in \citep{Zhou2014,Casale2015}) where
%
\begin{equation}
 \text{vec}(\mat{U}) \sim \multinormal N P {\text{vec}(\mat{0})} {\mat{C} \otimes \mat{K}}
 \label{eq:G-mn}
\end{equation}
%
The structure of \tmat{C} depends on the desired design of the covariance effect, which can be either shared or independent across traits. This distribution can be realised by simulation a random variable \(\mat{Z} \inR M L\) as iid \(\normal 0 1\) and setting 
\begin{equation}
 \text{vec}(\mat{U}) = \mat{BZA}^T
 \label{eq:G-sim}
\end{equation}
where \(\mat{B} \inR N M\) reflects the genetic relationship i.e. sample covariance with \(\mat{K}=\mat{BB}^T\) and \(\mat{A} \inR P L\) the trait covariance with  \(\mat{C}=\mat{AA}^T\), respectively (\(M\) and \(L\) depend on the rank of \(K\) and \(C\), hence are bound by \(N\) and \(P\)).  A detailed derivation for \cref{eq:G-sim} from \cref{eq:G-mn} can be found in \cref{section:simulating-G} and has similarly been applied in \citep{Casale2015}.

By recasting Equation~\ref{eq:G-mn} as Equation~\ref{eq:G-sim}, the infinitesimal genetic effect \tmat{U} described by a multivariate-normal distribution is effectively modelled as the product of three matrices, representing the sample covariance (\tmat{B}), a normally distributed variable (\tmat{Z}) and the trait covariance (\tmat{A}). Different designs of \tmat{A} will allow for the simulation of shared and independent genetic random effects. For the independent effect, \tmatsup{A}{ind} is a diagonal matrix with normally distributed entries: \((\matsup{A}{ind})^T = \text{diag}(a_1, a_2,  \dotsc , a_P) \sim \normal 0 1\), such that \(\matsup{U}{ind} =  \text{vec}(\mat{BZ}(\matsup{A}{ind})^T) \). \tmatsup{A}{shared} of the shared effect is simulated as a matrix of column rank one, with normally distributed entries in column one and zeros elsewhere: \(a_{i,1} \sim \normal 0 1\) and \(a_{i,j \neq 1} = 0\) such that \(\matsup{U}{shared} =  \text{vec}(\mat{BZ}(\matsup{A}{shared})^T) \).  
%
\item \textit{Correlated non-genetic effects:}  Correlated non-genetic effects are simulated as a multivariate normal distribution with a covariance matrix described by a defined trait-by-trait correlation. Any correlation structure between the phenotypes can be simulated with this effect component, as the desired correlation matrix \tmat{C} can be supplied by the user. In addition, as a simple approximation for spatially correlated phenotypes as they might occur for instance in image-based phenotypes, \textit{PhenotypeSimulator} provides the construction of \tmat{C} as follows: traits of distance \(d=1\) (adjacent trait columns) will have the highest specified correlation \(r\), traits with \(d=2\) have a correlation of \(r^2\), up to traits with \(d=(P - 1)\) with a correlation of \(r^{(P - 1)})\) , such that the correlation is highest at the first off-diagonal element and decreases exponentially by distance from the diagonal. The correlated non-genetic effect matrix is simulated as \(\mat{T} \sim \multinormal N P {\mat{0}} {\mat{C}}\).
%
\item \textit{Observational noise:} The observational noise effects \tmat{\Psi} are simulated as the sum of a shared and an independent observational noise effect. Both effect components are simulated by the matrix product of \(\mat{B} \inR N P   \sim \normal  0 1\) with \(\mat{A} \inR P P \). To realise the shared effect \tmatsup{\Psi}{shared}, \tmatsup{A}{shared} is simulated as a matrix of row rank one, with normally distributed entries in row one and zeros elsewhere: \(a_{1,j} \sim \normal 0 1\) and \(a_{i \neq 1,j} = 0\).  \tmat{A} of the independent component is a diagonal matrix with normally distributed entries:\\ \((\matsup{A}{ind})^T = \text{diag}(a_1, a_2,  \dotsc , a_P) \sim \normal 0 1\).
\end{enumerate}
%
\subsection{Scaling and phenotype construction}
\textit{PhenotypeSimulator} requires at least one phenotype component to simulate the phenotypes. Components can be combined as specified by the user and the correlation they introduce in the trait structure can be controlled by the specified levels of independent and shared effects (at the extremes, components can be simulated to either only have shared or independent effects). If desired, a simple phenotype structure following a model as cast for instance in the multi-variate normal model by \citep{Zhou2014} can be achieved by specifying only genetic variant effects, non-genetic covariate effects, infinitesimal genetic effects and observational noise.
%
I have designed \textit{PhenotypeSimulator} such that the amount of variance that each component should contribute to the total phenotypic variance can be specified by the user. Every component is thereby scaled by a factor \(a\) such that its average column variance explains \(x\) percent of the total variance. The scale factor \(a\) is derived as follows: 
Let \(X\) be a random variable with expected value \(E[X] = \mu_{x}\) and variance \(V[X] = E[(X - \mu_{x})^2]\) and let  \(Y = aX\). Then
  %
\begin{equation}
\begin{aligned}
E[Y] &= a\mu_{x} \\
V[Y] &= E[(Y - \mu_{y})^2] \\
V[Y] &= E[(aX - a\mu_{x})^2] \\
		&= a^2 E[(X - \mu_{x})^2]. \\
\end{aligned}
\end{equation}
%
Hence, the scaling of a random variable by \(a\) leads to the scaling of its variance by \(a^2\). To scale the phenotype components such that their average column variance \(\overline{V}_{col} = \frac{V_1 + ... + V_p}{p} \) explains a specified percentage~\(x\) of the total variance, choose the scaling factor \(a\) such that: 
\begin{equation}
\begin{aligned}
x  &= a^2 \times \overline{V}_{col} \\
a  &= \sqrt{x{\overline{V}_{col}}^{-1}}
\end{aligned}
\end{equation}
%
The final simulated phenotype \tmat{Y}  is expressed as the sum of the scaled genetic variant effects, the non-genetic covariates, the correlated non-genetic effects and observational noise effects:
\begin{equation}
\begin{aligned}
\mat{Y} &= \matsup{X}{shared}\matsup{B}{shared}  + \matsup{X}{ind}\matsup{B}{ind} + \matsup{W}{shared} \matsup{A}{shared} + \matsup{W}{ind}\matsup{A}{ind} \\
&+ \matsup{U}{shared} + \matsup{U}{ind} + \mat{T} + \matsup{\Psi}{shared} +  \matsup{\Psi}{ind} .
\end{aligned}
\end{equation}
%
\subsection{Case study}
To demonstrate the usage and application of \textit{PhenotypeSimulator}, I simulated a set of phenotypes and used them to evaluate the power of different linear mixed model designs in \gls{gwas}. In order to demonstrate the integration of \textit{PhenotypeSimulator} with already established simulation and \gls{gwas} tools, I choose Hapgen2 \citep{Su2011} for genotype simulation, used \textit{PhenotypeSimulator} for phenotype simulation based thereon and applied GEMMA (version 0.96) \citep{Zhou2014} for the \gls{gwas}. 
The analysis code and parameters of this case study, from the data simulation to the genome-wide association study are supplied as a vignette to the R package. 

I simulated genotype data for 1,000 individuals via Hagen2, mimicking population structure from four populations in the 1000Genomes project \citep{1000Genomes2012} (similar to the genotype structure described in \cref{section:genotype-simulation}). The simulated genotypes of this cohort served as the basis for the genetic variant and infinitesimal genetic effects. I generated a phenotype set consisting of three traits with ten genetic variant effects and four non-genetic covariates. For the ten genetic variant effects, I randomly selected ten variants from the genotypes and simulated shared genetic variant effects across all phenotypes. I introduced additional correlation structure by including an infinitesimal genetic effect based on the individuals' kinship estimates as well as a non-genetic correlated (correlation: \num{0.8}) and an observational noise effects. The total genetic variance accounts for \num{60}\% of the variance leaving \num{40}\% of variance explained by the noise terms. \Cref{fig:simulation} shows the trait-to-trait correlations of the final phenotype and each of its components.

\begin{figure}[h!]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.7\textwidth]{Chapter1/Figures/Phenotypes.pdf}
	\caption[\textbf{Phenotype simulation.}]{\textbf{Phenotype simulation.} Heatmaps of the trait-by-trait correlation (Pearson correlation) of a simulated phenotype ( \tmat{Y}) and its five phenotype components: genetic variant effects  \tmat{XB}, infinitesimal genetic effects \tmat{U}, non-genetic covariates \tmat{WA}, correlated non-genetic effects \tmat{T} and observational noise \tmat{\Psi}. The non-genetic covariates consist of four independent components, two following a binomial and two following a normal distribution. The genetic variant effect of ten causal \glspl{snp} with shared effect across all traits, yielding the strong correlation structure observed above. The highest correlation for the correlated non-genetic effect was set at \num{0.8}.}
	\label{fig:simulation}
\end{figure}

The final phenotypes served as the response variable in the \gls{gwas} based on \glspl{lmm} with the simulated \glspl{snp} and non-genetic covariates as fixed effects and the kinship estimated from the genotypes as part of the genetic random effect \citep{Zhou2014} (see \cref{subsection:lmm}). I analysed the power of jointly modelling all three phenotypes (multi-trait) and the power of single-trait models where the association of each phenotype is analysed separately. The single-trait \gls{gwas} was run for all three traits. All \gls{gwas} were conducted with GEMMA (version 0.96) \citep{Zhou2014}. In both, the multi-trait and single-trait \gls{gwas}, the phenotypes (-p flag) were modelled as the sum of genetic (simulated \glspl{snp}; -g flag) and non-genetic (simulated covariates; -c flag) fixed effects, a random genetic effect (with the eigenvectors and values of the kinship matrix, -u and -d flag) and observational noise (linear mixed model with likelihood ratio test using the -lmm 2 flag). For a comparison of the number of causal \glspl{snp} recovered in the multi-trait and single-trait \gls{gwas}, the p-values of the single-trait \gls{gwas} were adjusted by the number of test conducted (Bonferroni adjustment for three tests).

\begin{figure}[h]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.7\textwidth]{Chapter1/Figures/QQplot.pdf}
	\caption[\textbf{Comparison of multi-trait to single-trait GWAS.}]{\textbf{Comparison of multi-trait to single-trait GWAS.} Quantile-quantile plots of p-values observed from the multi-trait \gls{gwas} (via multivariate linear mixed model; mvLMM) to single-trait \gls{gwas} (via  univariate linear mixed models; uvLMM) fitted to each of the about eight million genome-wide SNPs (grey), including the ten SNPs for which a phenotype effect was modelled (green)}
	\label{fig:gwas-simulation}
\end{figure}

For the simulated phenotypes with shared genetic variant effects only, the multi-trait \gls{gwas} shows a greater power compared to any of the single trait analyses (\cref{fig:gwas-simulation}. The multi-trait \gls{gwas} detected four out of the ten \glspl{snp} for which a phenotype effect was modelled that pass the commonly used genome-wide significant threshold of \(5 \times 10^{-8}\) \citep{Fadista2016}. The single-trait \gls{gwas} only recovered three of these \glspl{snp}. The ability of linear (mixed) models to detect the \glspl{snp} for which a phenotype effect was modelled depends on the allele frequencies of these \glspl{snp} and the effect size \citep{Cohen1992,Halsey2015}: the higher the effect size and/or the allele frequencies the better the power to detect the \gls{snp} effects. The p-values of all \glspl{snp} with simulated effect on the phenotypes in relation to their allele frequencies and simulated effect sizes is depicted in \cref{fig:effect-pvalue-freq}. It shows a strong trend for \glspl{snp} with high allele frequencies and large simulated effect sizes to have low p-values. 

\begin{figure}[h]
	\centering
	\includegraphics[trim = 0mm 0mm 0mm 0mm, clip, width=0.7\textwidth]{Chapter1/Figures/effectsizes-freq-pvalues.pdf}
	\caption[\textbf{Relationship between p-values, allele frequencies and simulated effect sizes.}]{\textbf{Relationship between p-values, allele frequencies and simulated effect sizes.} The p-values of all \glspl{snp} with a simulated effect on the phenotypes are depicted in relation to their allele frequencies and simulated effect sizes. \glspl{snp} with low-allele frequencies and/or small simulated effect sizes do not pass the genome-wide significance threshold (horizontal line).}
	\label{fig:effect-pvalue-freq}
\end{figure}


\section{Conclusion}
\textit{PhenotypeSimulator} offers a framework for complex multi-trait, multi-locus phenotype simulations in quantitative genetics packaged in an easy to use manner for statistical geneticists.
\textit{PhenotypeSimulator} it is the only simulation package that I know that can simulate complex multi-trait phenotypes with complex multi-locus genetics, including a population structure term with phenotypic correlation. It  can create realistic covariate structures with similar properties (e.g. categorical covariates or covariates drawn from different distributions) to real covariates. The different phenotype components can be independently extracted and scaled, for example having the\textquote{true} variance components and covariance matrices from the simulation readily available for comparison to inference schemes. 

The underlying model for \textit{PhenotypeSimulator} corresponds to the common place linear mixed model framework. As such, it is limited in its use for benchmarking between methods, where linear mixed models methods are likely to perform best. However, the need for an underlying model is true for any simulation package.  

I have developed \textit{PhenotypeSimulator} as a flexible component in the standard genetics pipeline, with the ability to both read genetic formats from well used tools and output phenotypes compatible with many tools.  It is freely available as R/CRAN package and its code is present on github (\url{https://github.com/HannahVMeyer/PhenotypeSimulator}). This allows easy large scale deployment for comprehensive simulation across many parameter settings. 

In this thesis, phenotypes simulated with \textit{PhenotypeSimulator} built the basis for the method development in \cref{chapter:limmbo} and \cref{chapter:DimReduction}.